

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>OTUCLUST &mdash; micca 0.1 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '',
        VERSION:     '0.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="micca 0.1 documentation" href="index.html" />
    <link rel="prev" title="micca-test" href="command_ref/micca_test.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="command_ref/micca_test.html" title="micca-test"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">micca 0.1 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="otuclust">
<h1>OTUCLUST<a class="headerlink" href="#otuclust" title="Permalink to this headline">¶</a></h1>
<p>OTUCLUST is a new open-source application specifically created to
divide a set of amplicon reads into clusters. Given an identity
threshold (e.g. 97%), a cluster is defined by a centroid
(i.e. representative sequence ) and by the sequences which have
similarity above the threshold with the representative. OTUCLUST is
the default clustering method used in
<a class="reference internal" href="command_ref/micca_otu_denovo.html"><em>micca-otu-denovo</em></a>.</p>
<div class="section" id="definition-of-pair-wise-identity">
<h2>Definition of pair-wise identity<a class="headerlink" href="#definition-of-pair-wise-identity" title="Permalink to this headline">¶</a></h2>
<p>Different definitions of pair-wise identity are used by different
clustering software. There is no general consensus of how internal
gaps should be treated in this calculation. In OTUCLUST internal and
external gaps are ignored <a class="reference internal" href="#hwang2013">[Hwang2013]</a>. Given a global alignment (GA),
between the sequences <img class="math" src="_images/math/34857b3ba74ce5cd8607f3ebd23e9015908ada71.png" alt="i"/> and <img class="math" src="_images/math/8122aa89ea6e80784c6513d22787ad86e36ad0cc.png" alt="j"/> the pair-wise identity
is defined as:</p>
<div class="math">
<p><img src="_images/math/261b0f77589bb11c9fba60f241706f3267e6b59c.png" alt="s^{\text{GA}}_{i, j} = \left[\frac{\text{\# of matches}}{\text{\# of matches} +
    \text{\# of mismatches}} \text{of the GA}_{i,j}\right]"/></p>
</div><p>where mismatch and gap penalties equal to one.</p>
</div>
<div class="section" id="otuclust-algorithm-overview">
<h2>OTUCLUST algorithm overview<a class="headerlink" href="#otuclust-algorithm-overview" title="Permalink to this headline">¶</a></h2>
<p>The OTUCLUST procedure is composed by three main steps: <cite>a)</cite>
dereplication and abundance estimation, <cite>b)</cite> denovo chimera removal
(optional, with UCHIME) and <cite>c)</cite> clustering using the dereplicated
sequences as centroids.</p>
<div class="figure align-center">
<img alt="_images/otuclust.png" src="_images/otuclust.png" />
</div>
<div class="section" id="dereplication-and-chimera-removal-optional">
<h3>Dereplication and chimera removal (optional)<a class="headerlink" href="#dereplication-and-chimera-removal-optional" title="Permalink to this headline">¶</a></h3>
<p>The goal of the dereplication is the removal of duplicate
sequences. This step is performed by the clustering procedure
(described below) with an identity threshold of 100% or by exact
prefix matching (faster dereplication). At the end of the
dereplication step, singletons are discarded. Abundance is estimated
by counting the number of reads having a given unique sequence.
Dereplicated sequences are ordered by their abundance and passed to
UCHIME (denovo mode). Reads detected as chimeras are removed.</p>
</div>
<div class="section" id="clustering">
<h3>Clustering<a class="headerlink" href="#clustering" title="Permalink to this headline">¶</a></h3>
<p>Finally, dereplicated and chimera-free sequences are used as cluster
seeds by the clustering algorithm. The clustering procedure relies on
a <strong>search algorithm</strong> defined as follows:</p>
<ol class="arabic">
<li><p class="first">given a query sequence <img class="math" src="_images/math/9866e3a998d628ba0941eb4fea0666ac391d149a.png" alt="Q"/>, the sequence database is sorted by
decreasing k-mer similarity. The k-mer similarity is defined as
<a class="reference internal" href="#hwang2013">[Hwang2013]</a>:</p>
<div class="math">
<p><img src="_images/math/2bc06aeb55d58bbaa51abfc09930f0ebab0c5ae2.png" alt="s^{\text{k-mer}}_{i, j} = \frac{\sum_\tau \min(n_i(\tau),
    n_j(\tau))}{\min(l_i, l_j)-k+1}"/></p>
</div><p>where <img class="math" src="_images/math/1dc1c0119a604b91be9142370dc3159b6a9bbcb9.png" alt="\tau"/> is a k-mer, <img class="math" src="_images/math/174fadd07fd54c9afe288e96558c92e0c1da733a.png" alt="n"/> is the number of
occurences of the k-mer in the sequences and <img class="math" src="_images/math/bd76eb0d851c075e610e454c07fff42e5d697d85.png" alt="l_i, l_j"/> are
the lenght of sequences <img class="math" src="_images/math/34857b3ba74ce5cd8607f3ebd23e9015908ada71.png" alt="i"/> and <img class="math" src="_images/math/8122aa89ea6e80784c6513d22787ad86e36ad0cc.png" alt="j"/> respectively.</p>
</li>
<li><p class="first">For each sequence in the sorted database (target sequence,
<img class="math" src="_images/math/2554b6496c3b678897e9b060ef00aa9f0a7d7ece.png" alt="T"/>) the similarity <img class="math" src="_images/math/0f0dc3a7008b3c93db9462e43f8f939bd3079960.png" alt="s^{\text{GA}}_{Q, T}"/> is
computed. If <img class="math" src="_images/math/0f0dc3a7008b3c93db9462e43f8f939bd3079960.png" alt="s^{\text{GA}}_{Q, T}"/> is greater or equal than
the identity threshold <img class="math" src="_images/math/7b1a71954adb465aaf68dbf66c1c09cb51aef5f9.png" alt="s^{\text{thr}}"/> (e.g. 0.97) the
sequence <img class="math" src="_images/math/9866e3a998d628ba0941eb4fea0666ac391d149a.png" alt="Q"/> is added to the results and the reject counter
<img class="math" src="_images/math/61b858f7d680d02b795090a57e09217f6196c1ad.png" alt="n_{rej}"/> is set to zero. On the other hand <img class="math" src="_images/math/61b858f7d680d02b795090a57e09217f6196c1ad.png" alt="n_{rej}"/>
is incremented by one.</p>
</li>
<li><p class="first">If <img class="math" src="_images/math/61b858f7d680d02b795090a57e09217f6196c1ad.png" alt="n_{rej}"/> reaches the maximum number of consecutive
rejects allowed <img class="math" src="_images/math/f5047d1e0cbb50ec208923a22cd517c55100fa7b.png" alt="m"/> (e.g. 32) it is rather unlikely that an
other hit exist.</p>
</li>
</ol>
<p>Given a query sequence Q and the sequence database, the clustering
procedure searches (through the search algorithm) all the sequences in
the DB that have similarity above the identity threshold, forming a
new cluster. After that these sequences are removed from the sequence
database.</p>
</div>
</div>
<div class="section" id="otuclust-application">
<h2>OTUCLUST application<a class="headerlink" href="#otuclust-application" title="Permalink to this headline">¶</a></h2>
<div class="highlight-text"><div class="highlight"><pre>$ otuclust --help
usage: otuclust [-h] [-v] [-f {fastq,fasta}] [-s SIMILARITY] [-m SIZE] [-c]
                [-d] [-l LEN] [--out-clust FILE] [--out-rep FILE]
                input

otuclust - OTU clustering and chimera removal.

positional arguments:
  input                 input fastq/a file

optional arguments:
  -h, --help            show this help message and exit
  -v, --version         show program&#39;s version number and exit
  -f {fastq,fasta}, --format {fastq,fasta}
                        input file format (default fasta)
  -s SIMILARITY, --similarity SIMILARITY
                        similarity between cluster center and cluster
                        sequences (default 0.97)
  -m SIZE, --minsize SIZE
                        minimum size for a cluster (e.g. 2 removes singletons)
                        (default 2)
  -c, --remove-chimeras
                        remove chimeric sequences (recommended)
  -d, --derep-fast      fastest (prefix based) but less accurate dereplication
                        (recommended for dataset with 200000+ seqs)
  -l LEN, --derep-fast-len LEN
                        prefix length used in fast dereplication
  --out-clust FILE      output text file containing the clusters (default
                        clust.txt)
  --out-rep FILE        output fasta file containing the representative
                        sequences (default rep.fasta)

OTUCLUST 0.1
Author: Davide Albanese &lt;davide.albanese@fmach.it&gt;
Fondazione Edmund Mach, 2014.
</pre></div>
</div>
<p>Outputs:</p>
<dl class="docutils">
<dt><strong>clust.txt</strong></dt>
<dd><p class="first">a tab-delimited file where each row contains the sequence
identifiers assigned to the cluster. The first id corresponds to a
representative sequence. Sequence identifiers are coded as
<tt class="docutils literal"><span class="pre">SAMPLE_NAME||SEQ_ID</span></tt>:</p>
<div class="last highlight-python"><pre>sample1||F4HTPAO07H4B1Q sample1||F4HTPAO07ILHKH sample1||F4HTPAO07H8VJE  ...
sample3||F4HTPAO05FO0LC sample2||F4HTPAO02BVI74 sample3||F4HTPAO05FQCOF ...
...</pre>
</div>
</dd>
<dt><strong>rep.fasta</strong></dt>
<dd><p class="first">a FASTA file containing the representative sequence for each OTU:</p>
<div class="last highlight-python"><pre>&gt;sample1||F4HTPAO07H4B1Q
GTCCACGCCGTAAACGGTGGATGCTGGATGTGGGGCCCGTTCCACGGGTTCCGTGTCGGA
GCTAACGCGTTAAGCATCCCGCCTGGGGAGTACGGCCGCAAGGCTAAAACTCAAAGAAAT
TGACGGGGCCCGCACAAGCGGCGGAGCATGCGGATTAATTCGATGCAACGCGAAGAACCT
TACCTGGGCTTGACATGTTCCCGACGGTCGTAGAGATACGGCTTCCCTTCGGGGCGGGTT
CACAGGTGGTGCATGGTC
&gt;sample3||F4HTPAO05FO0LC
GTCCACGCCGTAAACGATGAATACTAGGTGTTGGGAAGCATTGCTTCTCGGTGCCGTCGC
AAACGCAGTAAGTATTCCACCTGGGGAGTACGTTCGCAAGAATGAAACTCAAAGGAATTG
ACGGGGACCCGCACAAGCGGTGGAGCATGTGGTTTAATTCGAAGCAACGCGAAGAACCTT
ACCAAGTCTTGACATCCTTCTGACCGGTACTTAACCGTACCTTCTCTTCGGAGCAGGAGT
GACAGGTGGTGCATGGTT
...</pre>
</div>
</dd>
</dl>
<table class="docutils citation" frame="void" id="hwang2013" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[Hwang2013]</td><td><em>(<a class="fn-backref" href="#id1">1</a>, <a class="fn-backref" href="#id2">2</a>)</em> Hwang et al. CLUSTOM: A Novel Method for Clustering 16S
rRNA Next Generation Sequences by Overlap Minimization.
PLoS ONE, 2013.</td></tr>
</tbody>
</table>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="index.html">
              <img class="logo" src="_static/micca_logo.png" alt="Logo"/>
            </a></p>
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">OTUCLUST</a><ul>
<li><a class="reference internal" href="#definition-of-pair-wise-identity">Definition of pair-wise identity</a></li>
<li><a class="reference internal" href="#otuclust-algorithm-overview">OTUCLUST algorithm overview</a><ul>
<li><a class="reference internal" href="#dereplication-and-chimera-removal-optional">Dereplication and chimera removal (optional)</a></li>
<li><a class="reference internal" href="#clustering">Clustering</a></li>
</ul>
</li>
<li><a class="reference internal" href="#otuclust-application">OTUCLUST application</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="command_ref/micca_test.html"
                        title="previous chapter"><tt class="docutils literal docutils literal docutils literal"><span class="pre">micca-test</span></tt></a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/otuclust.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="command_ref/micca_test.html" title="micca-test"
             >previous</a> |</li>
        <li><a href="index.html">micca 0.1 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2014, Davide Albanese, Paolo Fontana, Claudio Donati.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>