

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Import and analyze the data into R and phyloseq &mdash; micca 1.6.0 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="micca 1.6.0 documentation" href="index.html"/>
        <link rel="next" title="Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca" href="table.html"/>
        <link rel="prev" title="Paired-end sequencing" href="pairedend.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> micca
          

          
          </a>

          
            
            
              <div class="version">
                1.6
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="install.html">Install</a></li>
<li class="toctree-l1"><a class="reference internal" href="run.html">Run micca in 6 steps</a></li>
<li class="toctree-l1"><a class="reference internal" href="databases.html">Supported databases</a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="singleend.html">Single-end sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="pairedend.html">Paired-end sequencing</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Import and analyze the data into R and phyloseq</a></li>
<li class="toctree-l1"><a class="reference internal" href="table.html">Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="picrust.html">Picking OTUs for use in PICRUSt</a></li>
</ul>
<p class="caption"><span class="caption-text">In Depth</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="filtering.html">Quality filtering strategy in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="otu.html">OTU picking in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="formats.html">Supported file formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="changes.html">Changes</a></li>
</ul>
<p class="caption"><span class="caption-text">Command reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="commands/classify.html">classify</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/convert.html">convert</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/filter.html">filter</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/filterstats.html">filterstats</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/merge.html">merge</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/mergepairs.html">mergepairs</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/msa.html">msa</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/otu.html">otu</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/root.html">root</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/split.html">split</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/stats.html">stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tablebar.html">tablebar</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tablerare.html">tablerare</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tablestats.html">tablestats</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tabletotax.html">tabletotax</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tobiom.html">tobiom</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tree.html">tree</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/trim.html">trim</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">micca</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
      
    <li>Import and analyze the data into R and phyloseq</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/phyloseq.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="import-and-analyze-the-data-into-r-and-phyloseq">
<h1>Import and analyze the data into R and phyloseq<a class="headerlink" href="#import-and-analyze-the-data-into-r-and-phyloseq" title="Permalink to this headline">Â¶</a></h1>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This tutorial requires <a class="reference internal" href="singleend.html"><span class="doc">Single-end sequencing</span></a> to be done.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This tutorial requires <a class="reference external" href="https://www.r-project.org/">R</a> and
<a class="reference external" href="https://joey711.github.io/phyloseq/">phyloseq</a> (tested on R
v3.2.1 and phyloseq v1.14.0) to be installed in your system.</p>
</div>
<p>We can import the micca processed data (the BIOM file, the
phylogenetic tree and the representative sequences) into the <a class="reference external" href="https://www.r-project.org/">R</a> environment using the <a class="reference external" href="https://joey711.github.io/phyloseq/">phyloseq</a> library. From the phyloseq
homepage:</p>
<blockquote>
<div>&#8220;The phyloseq package is a tool to import, store, analyze, and
graphically display complex phylogenetic sequencing data that has
already been clustered into Operational Taxonomic Units (OTUs),
especially when there is associated sample data, phylogenetic tree,
and/or taxonomic assignment of the OTUs.&#8221;</div></blockquote>
<p>The <code class="docutils literal"><span class="pre">import_biom()</span></code> function allows to simultaneously import the BIOM
file and an associated phylogenetic tree file and reference sequence
file.</p>
<div class="highlight-R"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="kn">library</span><span class="p">(</span><span class="s">&quot;phyloseq&quot;</span><span class="p">)</span>
<span class="o">&gt;</span> <span class="kn">library</span><span class="p">(</span><span class="s">&quot;ggplot2&quot;</span><span class="p">)</span>
<span class="o">&gt;</span> theme_set<span class="p">(</span>theme_bw<span class="p">())</span>
<span class="o">&gt;</span> <span class="kp">setwd</span><span class="p">(</span><span class="s">&quot;denovo_greedy_otus&quot;</span><span class="p">)</span> <span class="c1"># set the working directory</span>
<span class="o">&gt;</span> ps <span class="o">=</span> import_biom<span class="p">(</span><span class="s">&quot;tables.biom&quot;</span><span class="p">,</span> treefilename<span class="o">=</span><span class="s">&quot;tree_rooted.tree&quot;</span><span class="p">,</span>
<span class="o">+</span> refseqfilename<span class="o">=</span><span class="s">&quot;otus.fasta&quot;</span><span class="p">)</span>
<span class="o">&gt;</span> ps
phyloseq<span class="o">-</span>class experiment<span class="o">-</span>level object
otu_table<span class="p">()</span>   OTU Table<span class="o">:</span>         <span class="p">[</span> <span class="m">1332</span> taxa and <span class="m">15</span> samples <span class="p">]</span>
sample_data<span class="p">()</span> Sample Data<span class="o">:</span>       <span class="p">[</span> <span class="m">15</span> samples by <span class="m">2</span> sample variables <span class="p">]</span>
tax_table<span class="p">()</span>   Taxonomy Table<span class="o">:</span>    <span class="p">[</span> <span class="m">1332</span> taxa by <span class="m">6</span> taxonomic ranks <span class="p">]</span>
phy_tree<span class="p">()</span>    Phylogenetic Tree<span class="o">:</span> <span class="p">[</span> <span class="m">1332</span> tips and <span class="m">1331</span> internal nodes <span class="p">]</span>
refseq<span class="p">()</span>      DNAStringSet<span class="o">:</span>      <span class="p">[</span> <span class="m">1332</span> reference sequences <span class="p">]</span>
</pre></div>
</div>
<p>At this point we can compute the number of OTUs and the Shannon
diversity index after the rarefaction:</p>
<div class="highlight-R"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="c1"># rarefy without replacement</span>
<span class="o">&gt;</span> ps.rarefied <span class="o">=</span> rarefy_even_depth<span class="p">(</span>ps<span class="p">,</span> rngseed<span class="o">=</span><span class="m">1</span><span class="p">,</span> sample.size<span class="o">=</span><span class="m">0.9</span><span class="o">*</span><span class="kp">min</span><span class="p">(</span>sample_sums<span class="p">(</span>ps<span class="p">)),</span> replace<span class="o">=</span><span class="bp">F</span><span class="p">)</span>
<span class="o">&gt;</span> <span class="c1"># plot alpha diversity indexes</span>
<span class="o">&gt;</span> plot_richness<span class="p">(</span>ps.rarefied<span class="p">,</span> x<span class="o">=</span><span class="s">&quot;Group&quot;</span><span class="p">,</span> measures<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="s">&quot;Observed&quot;</span><span class="p">,</span> <span class="s">&quot;Chao1&quot;</span><span class="p">,</span> <span class="s">&quot;Shannon&quot;</span><span class="p">))</span>
</pre></div>
</div>
<img alt="_images/alpha454.png" class="align-center" src="_images/alpha454.png" />
<p>Finnaly, we can plot the PCoA on the unweighted UniFrac distance of samples:</p>
<div class="highlight-R"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="c1"># PCoA plot on the unweighted UniFrac distance</span>
<span class="o">&gt;</span> ordination <span class="o">=</span> ordinate<span class="p">(</span>ps.rarefied<span class="p">,</span> method<span class="o">=</span><span class="s">&quot;PCoA&quot;</span><span class="p">,</span> distance<span class="o">=</span><span class="s">&quot;unifrac&quot;</span><span class="p">,</span> weighted<span class="o">=</span><span class="bp">F</span><span class="p">)</span>
<span class="o">&gt;</span> plot_ordination<span class="p">(</span>ps.rarefied<span class="p">,</span> ordination<span class="p">,</span> color<span class="o">=</span><span class="s">&quot;Group&quot;</span><span class="p">)</span> <span class="o">+</span> theme<span class="p">(</span>aspect.ratio<span class="o">=</span><span class="m">1</span><span class="p">)</span>
</pre></div>
</div>
<img alt="_images/beta454.png" class="align-center" src="_images/beta454.png" />
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="table.html" class="btn btn-neutral float-right" title="Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="pairedend.html" class="btn btn-neutral" title="Paired-end sequencing" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, micca development team.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'1.6.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>