

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca &mdash; micca 1.6.0 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="genindex.html"/>
        <link rel="search" title="Search" href="search.html"/>
    <link rel="top" title="micca 1.6.0 documentation" href="index.html"/>
        <link rel="next" title="Picking OTUs for use in PICRUSt" href="picrust.html"/>
        <link rel="prev" title="Import and analyze the data into R and phyloseq" href="phyloseq.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> micca
          

          
          </a>

          
            
            
              <div class="version">
                1.6
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="install.html">Install</a></li>
<li class="toctree-l1"><a class="reference internal" href="run.html">Run micca in 6 steps</a></li>
<li class="toctree-l1"><a class="reference internal" href="databases.html">Supported databases</a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="singleend.html">Single-end sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="pairedend.html">Paired-end sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="phyloseq.html">Import and analyze the data into R and phyloseq</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="picrust.html">Picking OTUs for use in PICRUSt</a></li>
</ul>
<p class="caption"><span class="caption-text">In Depth</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="filtering.html">Quality filtering strategy in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="otu.html">OTU picking in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="formats.html">Supported file formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="changes.html">Changes</a></li>
</ul>
<p class="caption"><span class="caption-text">Command reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="commands/classify.html">classify</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/convert.html">convert</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/filter.html">filter</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/filterstats.html">filterstats</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/merge.html">merge</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/mergepairs.html">mergepairs</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/msa.html">msa</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/otu.html">otu</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/root.html">root</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/split.html">split</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/stats.html">stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tablebar.html">tablebar</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tablerare.html">tablerare</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tablestats.html">tablestats</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tabletotax.html">tabletotax</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tobiom.html">tobiom</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tree.html">tree</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/trim.html">trim</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">micca</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
      
    <li>Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/table.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="compute-basic-otu-table-statistics-rarefy-and-summarize-otu-tables-by-taxa-using-micca">
<h1>Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca<a class="headerlink" href="#compute-basic-otu-table-statistics-rarefy-and-summarize-otu-tables-by-taxa-using-micca" title="Permalink to this headline">Â¶</a></h1>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This tutorial requires <a class="reference internal" href="singleend.html"><span class="doc">Single-end sequencing</span></a> to be done.</p>
</div>
<p>The command <a class="reference internal" href="commands/tablestats.html"><span class="doc">tablestats</span></a> reports a sample summary, an OTU summary and
the rarefaction curves for the input OTU table:</p>
<div class="highlight-sh"><div class="highlight"><pre><span></span>micca tablestats -i otutable.txt -o tablestats
</pre></div>
</div>
<p>Inspecting the file <code class="docutils literal"><span class="pre">tablestats/tablestats_samplesumm.txt</span></code> you can
see that the less abundant sample contains 9053 reads:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">Sample</span> <span class="n">Depth</span> <span class="n">NOTU</span> <span class="n">NSingle</span>
<span class="n">Mw_03</span>  <span class="mi">9053</span>  <span class="mi">716</span>  <span class="mi">142</span>
<span class="n">Mw_02</span>  <span class="mi">9947</span>  <span class="mi">760</span>  <span class="mi">166</span>
<span class="n">Mw_12</span>  <span class="mi">10843</span> <span class="mi">792</span>  <span class="mi">168</span>
<span class="o">...</span>    <span class="o">...</span>  <span class="o">...</span>   <span class="o">...</span>
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Rarefaction curves can be inspected through
<code class="docutils literal"><span class="pre">tablestats/tablestats_rarecurve.txt</span></code> and
<code class="docutils literal"><span class="pre">tablestats/tablestats_rarecurve_plot.png</span></code>.</p>
</div>
<p>To compare different samples, the OTU table must be subsampled
(<a class="reference external" href="https://en.wikipedia.org/wiki/Rarefaction_(ecology)">rarefied</a>)
using the command <a class="reference internal" href="commands/tablerare.html"><span class="doc">tablerare</span></a>. In this case we are
interested in rarefy the table with the depth of the less abundant
sample (<code class="docutils literal"><span class="pre">Mw_03</span></code>):</p>
<div class="highlight-sh"><div class="highlight"><pre><span></span>micca tablerare -i otutable.txt -o otutable_rare.txt -d <span class="m">9053</span>
</pre></div>
</div>
<p>Now we can summarize communities by their taxonomic composition.  The
<a class="reference internal" href="commands/tabletotax.html"><span class="doc">tabletotax</span></a> creates in the output directory a table for
each taxonomic level (<code class="docutils literal"><span class="pre">taxtable1.txt</span></code>, ..., <code class="docutils literal"><span class="pre">taxtableN.txt</span></code>). OTU
counts are summed together if they have the same taxonomy at the
considered level.</p>
<div class="highlight-sh"><div class="highlight"><pre><span></span>micca tabletotax -i otutable_rare.txt -t taxa.txt -o taxtables
</pre></div>
</div>
<p>Finally, we can generate a relative abundance bar plot from
generated taxa tables, using the command <a class="reference internal" href="commands/tablebar.html"><span class="doc">tablebar</span></a>. In
this case only the bar plot relative to the taxonomy level 2 (phylum)
will be generated:</p>
<div class="highlight-sh"><div class="highlight"><pre><span></span>micca tablebar -i taxtables/taxtable2.txt -o taxtables/taxtable2.png
</pre></div>
</div>
<img alt="_images/taxtable.png" class="align-center" src="_images/taxtable.png" />
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="picrust.html" class="btn btn-neutral float-right" title="Picking OTUs for use in PICRUSt" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="phyloseq.html" class="btn btn-neutral" title="Import and analyze the data into R and phyloseq" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, micca development team.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'1.6.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>