

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Single-end sequencing &mdash; micca 1.5.0 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
    <link rel="top" title="micca 1.5.0 documentation" href="index.html"/>
        <link rel="next" title="Paired-end sequencing" href="pairedend.html"/>
        <link rel="prev" title="Supported databases" href="databases.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> micca
          

          
          </a>

          
            
            
              <div class="version">
                1.5.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="install.html">Install</a></li>
<li class="toctree-l1"><a class="reference internal" href="run.html">Run micca in 6 steps</a></li>
<li class="toctree-l1"><a class="reference internal" href="databases.html">Supported databases</a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Single-end sequencing</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#dataset-download">Dataset download</a></li>
<li class="toctree-l2"><a class="reference internal" href="#merge-files">Merge files</a></li>
<li class="toctree-l2"><a class="reference internal" href="#primer-trimming">Primer trimming</a></li>
<li class="toctree-l2"><a class="reference internal" href="#quality-filtering">Quality filtering</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#choosing-parameters-for-filtering">Choosing parameters for filtering</a></li>
<li class="toctree-l3"><a class="reference internal" href="#filter-sequences">Filter sequences</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#otu-picking">OTU picking</a></li>
<li class="toctree-l2"><a class="reference internal" href="#assign-taxonomy">Assign taxonomy</a></li>
<li class="toctree-l2"><a class="reference internal" href="#infer-the-phylogenetic-tree">Infer the phylogenetic tree</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#multiple-sequence-alignment-msa">Multiple Sequence Alignment (MSA)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#build-the-phylogenetic-tree">Build the phylogenetic tree</a></li>
<li class="toctree-l3"><a class="reference internal" href="#midpoint-rooting">Midpoint rooting</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#build-the-biom-file">Build the BIOM file</a></li>
<li class="toctree-l2"><a class="reference internal" href="#import-and-analyze-the-data-into-r-and-phyloseq">Import and analyze the data into R and phyloseq</a></li>
<li class="toctree-l2"><a class="reference internal" href="#compute-basic-otu-table-statistics-rarefy-and-summarize-otu-tables-by-taxa-using-micca">Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="pairedend.html">Paired-end sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="phyloseq.html">Import and analyze the data into R and phyloseq</a></li>
<li class="toctree-l1"><a class="reference internal" href="table.html">Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="picrust.html">Picking OTUs for use in PICRUSt</a></li>
</ul>
<p class="caption"><span class="caption-text">In Depth</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="filtering.html">Quality filtering strategy in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="otu.html">OTU picking in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="formats.html">Supported file formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="changes.html">CHANGES</a></li>
</ul>
<p class="caption"><span class="caption-text">Command reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="commands/classify.html">classify</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/convert.html">convert</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/filter.html">filter</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/filterstats.html">filterstats</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/merge.html">merge</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/mergepairs.html">mergepairs</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/msa.html">msa</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/otu.html">otu</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/root.html">root</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/split.html">split</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/stats.html">stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tablebar.html">tablebar</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tablerare.html">tablerare</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tablestats.html">tablestats</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tabletotax.html">tabletotax</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tobiom.html">tobiom</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tree.html">tree</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/trim.html">trim</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">micca</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
      
    <li>Single-end sequencing</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/singleend.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="single-end-sequencing">
<h1>Single-end sequencing<a class="headerlink" href="#single-end-sequencing" title="Permalink to this headline">¶</a></h1>
<p>This tutorial describes a standard micca pipeline for the analysis
of single-end amplicon data. This pipeline is intended for different
platforms, such as <strong>Roche 454</strong>, <strong>Illumina MiSeq/HiSeq</strong> and <strong>Ion
Torrent</strong>. Although this tutorial explains how to apply the pipeline to
<strong>16S rRNA</strong> amplicons, it can be adapted to others markers gene/spacers,
e.g. <strong>Internal Transcribed Spacer (ITS)</strong> or <strong>28S</strong>.</p>
<div class="contents local topic" id="table-of-contents">
<p class="topic-title first">Table of Contents</p>
<ul class="simple">
<li><a class="reference internal" href="#dataset-download" id="id1">Dataset download</a></li>
<li><a class="reference internal" href="#merge-files" id="id2">Merge files</a></li>
<li><a class="reference internal" href="#primer-trimming" id="id3">Primer trimming</a></li>
<li><a class="reference internal" href="#quality-filtering" id="id4">Quality filtering</a></li>
<li><a class="reference internal" href="#otu-picking" id="id5">OTU picking</a></li>
<li><a class="reference internal" href="#assign-taxonomy" id="id6">Assign taxonomy</a></li>
<li><a class="reference internal" href="#infer-the-phylogenetic-tree" id="id7">Infer the phylogenetic tree</a></li>
<li><a class="reference internal" href="#build-the-biom-file" id="id8">Build the BIOM file</a></li>
<li><a class="reference internal" href="#import-and-analyze-the-data-into-r-and-phyloseq" id="id9">Import and analyze the data into R and phyloseq</a></li>
<li><a class="reference internal" href="#compute-basic-otu-table-statistics-rarefy-and-summarize-otu-tables-by-taxa-using-micca" id="id10">Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca</a></li>
</ul>
</div>
<div class="section" id="dataset-download">
<h2><a class="toc-backref" href="#id1">Dataset download</a><a class="headerlink" href="#dataset-download" title="Permalink to this headline">¶</a></h2>
<p>The dataset used in this tutorial is taken from the Barelli et
al. paper <em>Habitat fragmentation is associated to gut microbiota
diversity of an endangered primate: implications for conservation</em>
(<a class="reference external" href="https://doi.org/doi:10.1038/srep14862">https://doi.org/doi:10.1038/srep14862</a>). The dataset contains only a
subset of the entire study (Mwanihana samples only) for a total of
<strong>15 samples</strong> (in FASTQ format) and <strong>235179 16S rRNA amplicon
reads</strong> (V1-V3 hypervariable regions, 27-Forward
5&#8217;-AGAGTTTGATCMTGGCTCAG, 533-Reverse 5&#8217;-TTACCGCGGCTGCTGGCAC). The 454
pyrosequencing was carried out on the GS FLX+ system using the XL+
chemistry.</p>
<p>Open a terminal, download the data and prepare the working directory:</p>
<div class="highlight-sh"><div class="highlight"><pre><span></span>wget ftp://ftp.fmach.it/metagenomics/micca/examples/mwanihana.tar.gz
tar -zxvf mwanihana.tar.gz
<span class="nb">cd</span> mwanihana
</pre></div>
</div>
</div>
<div class="section" id="merge-files">
<h2><a class="toc-backref" href="#id2">Merge files</a><a class="headerlink" href="#merge-files" title="Permalink to this headline">¶</a></h2>
<p>Now the FASTQ files must be merged in a single file. This operation
can be performed with the <a class="reference internal" href="commands/merge.html"><span class="doc">merge</span></a> command. Sample names
will be included into the sequence indentifiers.</p>
<div class="highlight-sh"><div class="highlight"><pre><span></span>micca merge -i fastq/*.fastq -o merged.fastq
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The <a class="reference internal" href="commands/merge.html"><span class="doc">merge</span></a> command works with FASTQ or FASTA
files. If your sequences are in a different format (e.g. SFF or
FASTA+QUAL) use <a class="reference internal" href="commands/convert.html"><span class="doc">convert</span></a> to convert them.</p>
</div>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">In the case of multiplexed reads (with 5&#8217; barcode sequences) use
<a class="reference internal" href="commands/split.html"><span class="doc">split</span></a> instead of <a class="reference internal" href="commands/merge.html"><span class="doc">merge</span></a>. This
command will perform demultiplexing and merging at the same time.</p>
</div>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">In the case of overlapping paired-end reads go to
<a class="reference internal" href="pairedend.html"><span class="doc">Paired-end sequencing</span></a>.</p>
</div>
</div>
<div class="section" id="primer-trimming">
<span id="singleend-primer-trimming"></span><h2><a class="toc-backref" href="#id3">Primer trimming</a><a class="headerlink" href="#primer-trimming" title="Permalink to this headline">¶</a></h2>
<p>Segments which match PCR primers should be now removed. Typical Roche
454 reads start with a sequence key (e.g. TCAG) followed by the
barcode (if it was not previously removed) and the forward primer.
For these types of data (and in general, for single-end sequencing) we
recommend to <strong>trim both forward reverse primers and discard reads that
do not contain the forward primer</strong>. Moreover, sequence preceding (for
the forward) or succeding (for the reverse, if found) primers should
be removed:</p>
<img alt="_images/read454.png" class="align-center" src="_images/read454.png" />
<p>These operations can be performed with the <a class="reference internal" href="commands/trim.html"><span class="doc">trim</span></a> command:</p>
<div class="highlight-sh"><div class="highlight"><pre><span></span>micca trim -i merged.fastq -o trimmed.fastq -w AGAGTTTGATCMTGGCTCAG -r GTGCCAGCAGCCGCGGTAA -W
</pre></div>
</div>
<p>The option <code class="docutils literal"><span class="pre">-W/--duforward</span></code> ensures that reads that do not contain
the forward primer will be discarded.</p>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">Do not use the <code class="docutils literal"><span class="pre">-R/--dureverse</span></code> with single-end reads.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The <a class="reference internal" href="commands/trim.html"><span class="doc">trim</span></a> command supports <a class="reference external" href="http://www.bioinformatics.org/sms/iupac.html">IUPAC</a> nucleotide codes
and multiple primers. With the option <code class="docutils literal"><span class="pre">-c/--searchrc</span></code> the command
searches reverse complement primers too. <a class="reference internal" href="commands/trim.html"><span class="doc">trim</span></a>
works with FASTQ or FASTA files.</p>
</div>
</div>
<div class="section" id="quality-filtering">
<span id="singleend-quality-filtering"></span><h2><a class="toc-backref" href="#id4">Quality filtering</a><a class="headerlink" href="#quality-filtering" title="Permalink to this headline">¶</a></h2>
<p>Producing high-quality OTUs requires high-quality reads. The
<a class="reference internal" href="commands/filter.html"><span class="doc">filter</span></a> command filters sequences according to the
maximum allowed expected error (EE) rate % (see <a class="reference internal" href="filtering.html"><span class="doc">Quality filtering strategy in micca</span></a>). We
recommend values &lt;=1%. Moreover, to obtain good results in clustering
(see <a class="reference internal" href="commands/otu.html"><span class="doc">otu</span></a>), reads should be <strong>truncated at the same
length</strong> when they cover partial amplicons or if quality deteriorates
towards the end (common when you have long amplicons in single-end
sequencing).</p>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">Parameters for the <a class="reference internal" href="commands/filter.html"><span class="doc">filter</span></a> command should be chosen
using the command <a class="reference internal" href="commands/filterstats.html"><span class="doc">filterstats</span></a>.</p>
</div>
<div class="section" id="choosing-parameters-for-filtering">
<h3>Choosing parameters for filtering<a class="headerlink" href="#choosing-parameters-for-filtering" title="Permalink to this headline">¶</a></h3>
<p>The command <a class="reference internal" href="commands/filterstats.html"><span class="doc">filterstats</span></a> reports the fraction of reads
that would pass for each specified maximum expected error (EE) rate %:</p>
<div class="highlight-sh"><div class="highlight"><pre><span></span>micca filterstats -i trimmed.fastq -o filterstats
</pre></div>
</div>
<p>Open the PNG file <code class="docutils literal"><span class="pre">filterstats/stats_plot.png</span></code>:</p>
<img alt="_images/filterstats454.png" class="align-center" src="_images/filterstats454.png" />
<p>In this case we are interested in the plot below (minimum length
filtering + truncation). A truncation length of <strong>350</strong> and a maximum
error rate of <strong>0.5%</strong> seems to be a good compromise between read read
length, expected error rate and number of reads remaining. Inspecting
the file <code class="docutils literal"><span class="pre">filterstats/trunclen_stats.txt</span></code>, you can see that more
than <strong>92%</strong> reads will pass the filter:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">L</span>       <span class="mf">0.25</span>    <span class="mf">0.5</span>     <span class="mf">0.75</span>    <span class="mf">1.0</span>     <span class="mf">1.25</span>    <span class="mf">1.5</span>
<span class="o">...</span>
<span class="mi">349</span>     <span class="mf">78.905</span>  <span class="mf">92.472</span>  <span class="mf">97.425</span>  <span class="mf">99.135</span>  <span class="mf">99.705</span>  <span class="mf">99.897</span>
<span class="mi">350</span>     <span class="mf">78.639</span>  <span class="mf">92.385</span>  <span class="mf">97.389</span>  <span class="mf">99.126</span>  <span class="mf">99.704</span>  <span class="mf">99.896</span>
<span class="mi">351</span>     <span class="mf">78.369</span>  <span class="mf">92.300</span>  <span class="mf">97.357</span>  <span class="mf">99.116</span>  <span class="mf">99.700</span>  <span class="mf">99.892</span>
<span class="o">...</span>
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">To obtain general sequencing statistics, run
<a class="reference internal" href="commands/stats.html"><span class="doc">stats</span></a>.</p>
</div>
</div>
<div class="section" id="filter-sequences">
<h3>Filter sequences<a class="headerlink" href="#filter-sequences" title="Permalink to this headline">¶</a></h3>
<p>Now we can run the <a class="reference internal" href="commands/filter.html"><span class="doc">filter</span></a> command with
the selected parameters:</p>
<div class="highlight-sh"><div class="highlight"><pre><span></span>micca filter -i trimmed.fastq -o filtered.fasta -e 0.5 -m <span class="m">350</span> -t
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The maximum number of allowed Ns after truncation can be also
specified in <a class="reference internal" href="commands/filterstats.html"><span class="doc">filterstats</span></a> and in
<a class="reference internal" href="commands/filter.html"><span class="doc">filter</span></a>.</p>
</div>
</div>
</div>
<div class="section" id="otu-picking">
<span id="singleend-otu-picking"></span><h2><a class="toc-backref" href="#id5">OTU picking</a><a class="headerlink" href="#otu-picking" title="Permalink to this headline">¶</a></h2>
<p>To characterize the taxonomic structure of the samples, the sequences
are now organized into <a class="reference external" href="https://en.wikipedia.org/wiki/Operational_taxonomic_unit">Operational Taxonomic Units (OTUs)</a> at
varying levels of identity. An identity of <strong>97%</strong> represent the
common working definition of bacterial species. The
<a class="reference internal" href="commands/otu.html"><span class="doc">otu</span></a> command implements several state-of-the-art
approaches for OTU clustering, but in this tutorial we will focus on
the <strong>de novo greedy clustering</strong> (see <a class="reference internal" href="otu.html"><span class="doc">OTU picking in micca</span></a>):</p>
<div class="highlight-sh"><div class="highlight"><pre><span></span>micca otu -i filtered.fasta -o denovo_greedy_otus -d 0.97 -c -t 4
</pre></div>
</div>
<p>The <a class="reference internal" href="commands/otu.html"><span class="doc">otu</span></a> command returns several files in the output
directory, including the <strong>OTU table</strong> (<code class="docutils literal"><span class="pre">otutable.txt</span></code>) and a FASTA
file containing the <strong>representative sequences</strong> (<code class="docutils literal"><span class="pre">otus.fasta</span></code>).</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">See <a class="reference internal" href="otu.html"><span class="doc">OTU picking in micca</span></a> to see how to apply the <strong>swarm de novo</strong>,
<strong>closed-reference</strong> and the <strong>open-reference</strong> OTU picking strategies to
these data.</p>
</div>
</div>
<div class="section" id="assign-taxonomy">
<h2><a class="toc-backref" href="#id6">Assign taxonomy</a><a class="headerlink" href="#assign-taxonomy" title="Permalink to this headline">¶</a></h2>
<p>Now we can assign taxonomy to each representative sequence using the
<a class="reference internal" href="commands/classify.html"><span class="doc">classify</span></a> command. In this tutorial we use the RDP
(<a class="reference external" href="https://doi.org/10.1128/AEM.00062-07">https://doi.org/10.1128/AEM.00062-07</a>) classifier.</p>
<div class="highlight-sh"><div class="highlight"><pre><span></span>micca classify -m rdp -i denovo_greedy_otus/otus.fasta -o denovo_greedy_otus/taxa.txt
</pre></div>
</div>
<p><a class="reference internal" href="commands/classify.html"><span class="doc">classify</span></a> returns a taxonomy file like this:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">DENOVO1</span> <span class="n">Bacteria</span><span class="p">;</span><span class="n">Firmicutes</span><span class="p">;</span><span class="n">Clostridia</span><span class="p">;</span><span class="n">Clostridiales</span>
<span class="n">DENOVO2</span> <span class="n">Bacteria</span><span class="p">;</span><span class="n">Bacteroidetes</span><span class="p">;</span><span class="n">Bacteroidia</span><span class="p">;</span><span class="n">Bacteroidales</span>
<span class="n">DENOVO3</span> <span class="n">Bacteria</span><span class="p">;</span><span class="n">Verrucomicrobia</span><span class="p">;</span><span class="n">Opitutae</span>
<span class="n">DENOVO4</span> <span class="n">Bacteria</span><span class="p">;</span><span class="n">Bacteroidetes</span><span class="p">;</span><span class="n">Bacteroidia</span><span class="p">;</span><span class="n">Bacteroidales</span>
<span class="o">...</span>
</pre></div>
</div>
</div>
<div class="section" id="infer-the-phylogenetic-tree">
<h2><a class="toc-backref" href="#id7">Infer the phylogenetic tree</a><a class="headerlink" href="#infer-the-phylogenetic-tree" title="Permalink to this headline">¶</a></h2>
<p>These steps are necessary if you want to use phylogenetic-based
metrics such as the UniFrac distance
(<a class="reference external" href="https://doi.org/10.1128/AEM.01996-06">https://doi.org/10.1128/AEM.01996-06</a>) in the downstream analysis.</p>
<div class="section" id="multiple-sequence-alignment-msa">
<h3>Multiple Sequence Alignment (MSA)<a class="headerlink" href="#multiple-sequence-alignment-msa" title="Permalink to this headline">¶</a></h3>
<p>The <a class="reference internal" href="commands/msa.html"><span class="doc">msa</span></a> command provides two approaches for <a class="reference external" href="https://en.wikipedia.org/wiki/Multiple_sequence_alignment">MSA</a>: MUSCLE
(<a class="reference external" href="https://doi.org/10.1093/nar/gkh340">https://doi.org/10.1093/nar/gkh340</a>) (de novo alignment) and Nearest
Alignment Space Termination (NAST)
(<a class="reference external" href="https://doi.org/10.1093/nar/gkl244">https://doi.org/10.1093/nar/gkl244</a>) (which uses a template
alignment). In this tutorial we will use the NAST alignment
method. For 16S rRNA sequences, a good template alignment is the
Greengenes Core Set:</p>
<div class="highlight-sh"><div class="highlight"><pre><span></span>wget ftp://ftp.fmach.it/metagenomics/micca/dbs/core_set.tar.gz
tar -zxvf core_set.tar.gz
</pre></div>
</div>
<p>At this point we can run the <a class="reference internal" href="commands/msa.html"><span class="doc">msa</span></a> command:</p>
<div class="highlight-sh"><div class="highlight"><pre><span></span>micca msa -m nast -i denovo_greedy_otus/otus.fasta -o denovo_greedy_otus/msa.fasta --nast-template core_set_aligned.fasta.imputed --nast-threads 4
</pre></div>
</div>
</div>
<div class="section" id="build-the-phylogenetic-tree">
<h3>Build the phylogenetic tree<a class="headerlink" href="#build-the-phylogenetic-tree" title="Permalink to this headline">¶</a></h3>
<p>At this point we can build the phylogenetic tree from the MSA using
<a class="reference internal" href="commands/tree.html"><span class="doc">tree</span></a>:</p>
<div class="highlight-sh"><div class="highlight"><pre><span></span>micca tree -i denovo_greedy_otus/msa.fasta -o denovo_greedy_otus/tree.tree
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The output tree is in <a class="reference external" href="https://en.wikipedia.org/wiki/Newick_format">Newick format</a>.</p>
</div>
</div>
<div class="section" id="midpoint-rooting">
<h3>Midpoint rooting<a class="headerlink" href="#midpoint-rooting" title="Permalink to this headline">¶</a></h3>
<p>UniFrac metrics require phylogenetic trees to be rooted. The tree can
be rooted (in this case at midpoint between the two most distant tips
of the tree) using the <a class="reference internal" href="commands/root.html"><span class="doc">root</span></a> command:</p>
<div class="highlight-sh"><div class="highlight"><pre><span></span>micca root -i denovo_greedy_otus/tree.tree -o denovo_greedy_otus/tree_rooted.tree
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Tree can also be rooted with the outgroup clade containing selected
targets, see <a class="reference internal" href="commands/root.html"><span class="doc">root</span></a>.</p>
</div>
</div>
</div>
<div class="section" id="build-the-biom-file">
<h2><a class="toc-backref" href="#id8">Build the BIOM file</a><a class="headerlink" href="#build-the-biom-file" title="Permalink to this headline">¶</a></h2>
<p>The <a class="reference external" href="http://biom-format.org/">Biological Observation Matrix (BIOM)</a>
is a common format for representing OTU tables and metadata and is the
core data type for downstream analyses in <a class="reference external" href="http://qiime.org">QIIME</a>
and in <a class="reference external" href="https://joey711.github.io/phyloseq/">phyloseq</a>. <a class="reference internal" href="commands/tobiom.html"><span class="doc">tobiom</span></a>
converts the OTU table and the taxonomy table produced by the previous
steps to the BIOM format. In addition, the <a class="reference internal" href="formats.html#formats-sample-data"><span class="std std-ref">Sample data</span></a> can
be added:</p>
<div class="highlight-sh"><div class="highlight"><pre><span></span>micca tobiom -i denovo_greedy_otus/otutable.txt -o denovo_greedy_otus/tables.biom -t denovo_greedy_otus/taxa.txt -s sampledata.txt
</pre></div>
</div>
</div>
<div class="section" id="import-and-analyze-the-data-into-r-and-phyloseq">
<h2><a class="toc-backref" href="#id9">Import and analyze the data into R and phyloseq</a><a class="headerlink" href="#import-and-analyze-the-data-into-r-and-phyloseq" title="Permalink to this headline">¶</a></h2>
<p>See <a class="reference internal" href="phyloseq.html"><span class="doc">Import and analyze the data into R and phyloseq</span></a></p>
</div>
<div class="section" id="compute-basic-otu-table-statistics-rarefy-and-summarize-otu-tables-by-taxa-using-micca">
<h2><a class="toc-backref" href="#id10">Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca</a><a class="headerlink" href="#compute-basic-otu-table-statistics-rarefy-and-summarize-otu-tables-by-taxa-using-micca" title="Permalink to this headline">¶</a></h2>
<p>See <a class="reference internal" href="table.html"><span class="doc">Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca</span></a></p>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="pairedend.html" class="btn btn-neutral float-right" title="Paired-end sequencing" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="databases.html" class="btn btn-neutral" title="Supported databases" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2016, micca development team.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'1.5.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>