

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>classify &mdash; micca 1.6.0 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="../genindex.html"/>
        <link rel="search" title="Search" href="../search.html"/>
    <link rel="top" title="micca 1.6.0 documentation" href="../index.html"/>
        <link rel="next" title="convert" href="convert.html"/>
        <link rel="prev" title="Changes" href="../changes.html"/> 

  
  <script src="../_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../index.html" class="icon icon-home"> micca
          

          
          </a>

          
            
            
              <div class="version">
                1.6
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../install.html">Install</a></li>
<li class="toctree-l1"><a class="reference internal" href="../run.html">Run micca in 6 steps</a></li>
<li class="toctree-l1"><a class="reference internal" href="../databases.html">Supported databases</a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../singleend.html">Single-end sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../pairedend.html">Paired-end sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../phyloseq.html">Import and analyze the data into R and phyloseq</a></li>
<li class="toctree-l1"><a class="reference internal" href="../table.html">Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="../picrust.html">Picking OTUs for use in PICRUSt</a></li>
</ul>
<p class="caption"><span class="caption-text">In Depth</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../filtering.html">Quality filtering strategy in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otu.html">OTU picking in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="../formats.html">Supported file formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="../changes.html">Changes</a></li>
</ul>
<p class="caption"><span class="caption-text">Command reference</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">classify</a></li>
<li class="toctree-l1"><a class="reference internal" href="convert.html">convert</a></li>
<li class="toctree-l1"><a class="reference internal" href="filter.html">filter</a></li>
<li class="toctree-l1"><a class="reference internal" href="filterstats.html">filterstats</a></li>
<li class="toctree-l1"><a class="reference internal" href="merge.html">merge</a></li>
<li class="toctree-l1"><a class="reference internal" href="mergepairs.html">mergepairs</a></li>
<li class="toctree-l1"><a class="reference internal" href="msa.html">msa</a></li>
<li class="toctree-l1"><a class="reference internal" href="otu.html">otu</a></li>
<li class="toctree-l1"><a class="reference internal" href="root.html">root</a></li>
<li class="toctree-l1"><a class="reference internal" href="split.html">split</a></li>
<li class="toctree-l1"><a class="reference internal" href="stats.html">stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="tablebar.html">tablebar</a></li>
<li class="toctree-l1"><a class="reference internal" href="tablerare.html">tablerare</a></li>
<li class="toctree-l1"><a class="reference internal" href="tablestats.html">tablestats</a></li>
<li class="toctree-l1"><a class="reference internal" href="tabletotax.html">tabletotax</a></li>
<li class="toctree-l1"><a class="reference internal" href="tobiom.html">tobiom</a></li>
<li class="toctree-l1"><a class="reference internal" href="tree.html">tree</a></li>
<li class="toctree-l1"><a class="reference internal" href="trim.html">trim</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../index.html">micca</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../index.html">Docs</a> &raquo;</li>
      
    <li>classify</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../_sources/commands/classify.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="classify">
<h1>classify<a class="headerlink" href="#classify" title="Permalink to this headline">Â¶</a></h1>
<div class="highlight-console"><div class="highlight"><pre><span></span><span class="go">usage: micca classify [-h] -i FILE -o FILE [-m {cons,rdp,otuid}] [-r FILE]</span>
<span class="go">                    [-x FILE] [--cons-id CONS_ID]</span>
<span class="go">                    [--cons-maxhits CONS_MAXHITS]</span>
<span class="go">                    [--cons-minfrac CONS_MINFRAC]</span>
<span class="go">                    [--cons-mincov CONS_MINCOV] [--cons-strand {both,plus}]</span>
<span class="go">                    [--cons-threads THREADS]</span>
<span class="go">                    [--rdp-gene {16srrna,fungallsu,fungalits_warcup,fungalits_unite}]</span>
<span class="go">                    [--rdp-maxmem GB] [--rdp-minconf RDP_MINCONF]</span>

<span class="go">micca classify assigns taxonomy for each sequence in the input file</span>
<span class="go">and provides three methods for classification:</span>

<span class="go">* VSEARCH-based consensus classifier (cons): input sequences are</span>
<span class="go">searched in the reference database with VSEARCH</span>
<span class="go">(https://github.com/torognes/vsearch). For each query sequence the</span>
<span class="go">method retrives up to &#39;cons-maxhits&#39; hits (i.e. identity &gt;=</span>
<span class="go">&#39;cons-id&#39;). Then, the most specific taxonomic label that is</span>
<span class="go">associated with at least &#39;cons-minfrac&#39; of the hits is</span>
<span class="go">assigned. The method is similar to the UCLUST-based consensus</span>
<span class="go">taxonomy assigner presented in doi: 10.7287/peerj.preprints.934v2</span>
<span class="go">and available in QIIME.</span>

<span class="go">* RDP classifier (rdp): only RDP classifier version &gt;= 2.8 is</span>
<span class="go">supported (doi:10.1128/AEM.00062-07). In order to use this</span>
<span class="go">classifier RDP must be installed (download at</span>
<span class="go">http://sourceforge.net/projects/rdp-classifier/files/rdp-classifier/)</span>
<span class="go">and the RDPPATH environment variable setted. The available</span>
<span class="go">databases (--rdp-gene) are:</span>

<span class="go">- 16S (16srrna)</span>
<span class="go">- Fungal LSU (28S) (fungallsu)</span>
<span class="go">- Warcup ITS (fungalits_warcup, doi: 10.3852/14-293)</span>
<span class="go">- UNITE ITS (fungalits_unite)</span>

<span class="go">For more information about the RDP classifier go to</span>
<span class="go">http://rdp.cme.msu.edu/classifier/classifier.jsp</span>

<span class="go">* OTU ID classifier (otuid): simply perform a sequence ID matching</span>
<span class="go">with the reference taxonomy file. Recommended strategy when the</span>
<span class="go">closed reference clustering (--method closedref in micca-otu) was</span>
<span class="go">performed. OTU ID classifier requires a tab-delimited file where</span>
<span class="go">the first column contains the current OTU ids and the second column</span>
<span class="go">the reference taxonomy ids (see otuids.txt in micca-otu), e.g.:</span>

<span class="go">REF1[TAB]1110191</span>
<span class="go">REF2[TAB]1104777</span>
<span class="go">REF3[TAB]1078527</span>
<span class="go">...</span>

<span class="go">The input reference taxonomy file (--ref-tax) should be a</span>
<span class="go">tab-delimited file where rows are either in the form:</span>

<span class="go">1. SEQID[TAB]k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__;g__;</span>
<span class="go">2. SEQID[TAB]Bacteria;Firmicutes;Clostridia;Clostridiales;;;</span>
<span class="go">3. SEQID[TAB]Bacteria;Firmicutes;Clostridia;Clostridiales</span>
<span class="go">4. SEQID[TAB]D_0__Bacteria;D_1__Firmicutes;D_2__Clostridia;D_3__Clostridiales;D_4__;D_5__;</span>

<span class="go">Compatible reference database are Greengenes</span>
<span class="go">(http://greengenes.secondgenome.com/downloads), QIIME-formatted SILVA</span>
<span class="go">(https://www.arb-silva.de/download/archive/qiime/) and UNITE</span>
<span class="go">(https://unite.ut.ee/repository.php).</span>

<span class="go">The output file is a tab-delimited file where each row is in the</span>
<span class="go">format:</span>

<span class="go">SEQID[TAB]Bacteria;Firmicutes;Clostridia;Clostridiales</span>

<span class="go">optional arguments:</span>
<span class="go">-h, --help            show this help message and exit</span>

<span class="go">arguments:</span>
<span class="go">-i FILE, --input FILE</span>
<span class="go">                        input FASTA file (for &#39;cons&#39; and &#39;rdp&#39;) or a tab-</span>
<span class="go">                        delimited OTU ids file (for &#39;otuid&#39;) (required).</span>
<span class="go">-o FILE, --output FILE</span>
<span class="go">                        output taxonomy file (required).</span>
<span class="go">-m {cons,rdp,otuid}, --method {cons,rdp,otuid}</span>
<span class="go">                        classification method (default cons)</span>
<span class="go">-r FILE, --ref FILE   reference sequences in FASTA format, required for</span>
<span class="go">                        &#39;cons&#39; classifier.</span>
<span class="go">-x FILE, --ref-tax FILE</span>
<span class="go">                        tab-separated reference taxonomy file, required for</span>
<span class="go">                        &#39;cons&#39; and &#39;otuid&#39; classifiers.</span>

<span class="go">VSEARCH-based consensus classifierspecific options:</span>
<span class="go">--cons-id CONS_ID     sequence identity threshold (0.0 to 1.0, default 0.9).</span>
<span class="go">--cons-maxhits CONS_MAXHITS</span>
<span class="go">                        number of hits to consider (&gt;=1, default 3).</span>
<span class="go">--cons-minfrac CONS_MINFRAC</span>
<span class="go">                        for each taxonomic rank, a specific taxa will be</span>
<span class="go">                        assigned if it is present in at least MINFRAC of the</span>
<span class="go">                        hits (0.0 to 1.0, default 0.5).</span>
<span class="go">--cons-mincov CONS_MINCOV</span>
<span class="go">                        reject sequence if the fraction of alignment to the</span>
<span class="go">                        reference sequence is lower than MINCOV. This</span>
<span class="go">                        parameter prevents low-coverage alignments at the end</span>
<span class="go">                        of the sequences (default 0.75).</span>
<span class="go">--cons-strand {both,plus}</span>
<span class="go">                        search both strands or the plus strand only (default</span>
<span class="go">                        both).</span>
<span class="go">--cons-threads THREADS</span>
<span class="go">                        number of threads to use (1 to 256, default 1).</span>

<span class="go">RDP Classifier/Database specific options:</span>
<span class="go">--rdp-gene {16srrna,fungallsu,fungalits_warcup,fungalits_unite}</span>
<span class="go">                        marker gene/region</span>
<span class="go">--rdp-maxmem GB       maximum memory size for the java virtual machine in GB</span>
<span class="go">                        (default 2)</span>
<span class="go">--rdp-minconf RDP_MINCONF</span>
<span class="go">                        minimum confidence value to assign taxonomy to a</span>
<span class="go">                        sequence (default 0.8)</span>

<span class="go">Examples</span>

<span class="go">Classification of 16S sequences using the consensus classifier and</span>
<span class="go">Greengenes:</span>

<span class="go">    micca classify -m cons -i input.fasta -o tax.txt \</span>
<span class="go">    --ref greengenes_2013_05/rep_set/97_otus.fasta \</span>
<span class="go">    --ref-tax greengenes_2013_05/taxonomy/97_otu_taxonomy.txt</span>

<span class="go">Classification of ITS sequences using the RDP classifier and the</span>
<span class="go">UNITE database:</span>

<span class="go">    micca classify -m rdp --rdp-gene fungalits_unite -i input.fasta \</span>
<span class="go">    -o tax.txt</span>

<span class="go">OTU ID matching after the closed reference OTU picking protocol:</span>

<span class="go">    micca classify -m otuid -i otuids.txt -o tax.txt \</span>
<span class="go">    --ref-tax greengenes_2013_05/taxonomy/97_otu_taxonomy.txt</span>
</pre></div>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="convert.html" class="btn btn-neutral float-right" title="convert" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../changes.html" class="btn btn-neutral" title="Changes" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, micca development team.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'1.6.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>