

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>msa &mdash; micca 1.6.0 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="../genindex.html"/>
        <link rel="search" title="Search" href="../search.html"/>
    <link rel="top" title="micca 1.6.0 documentation" href="../index.html"/>
        <link rel="next" title="otu" href="otu.html"/>
        <link rel="prev" title="mergepairs" href="mergepairs.html"/> 

  
  <script src="../_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../index.html" class="icon icon-home"> micca
          

          
          </a>

          
            
            
              <div class="version">
                1.6
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../install.html">Install</a></li>
<li class="toctree-l1"><a class="reference internal" href="../run.html">Run micca in 6 steps</a></li>
<li class="toctree-l1"><a class="reference internal" href="../databases.html">Supported databases</a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../singleend.html">Single-end sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../pairedend.html">Paired-end sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../phyloseq.html">Import and analyze the data into R and phyloseq</a></li>
<li class="toctree-l1"><a class="reference internal" href="../table.html">Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="../picrust.html">Picking OTUs for use in PICRUSt</a></li>
</ul>
<p class="caption"><span class="caption-text">In Depth</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../filtering.html">Quality filtering strategy in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otu.html">OTU picking in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="../formats.html">Supported file formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="../changes.html">Changes</a></li>
</ul>
<p class="caption"><span class="caption-text">Command reference</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="classify.html">classify</a></li>
<li class="toctree-l1"><a class="reference internal" href="convert.html">convert</a></li>
<li class="toctree-l1"><a class="reference internal" href="filter.html">filter</a></li>
<li class="toctree-l1"><a class="reference internal" href="filterstats.html">filterstats</a></li>
<li class="toctree-l1"><a class="reference internal" href="merge.html">merge</a></li>
<li class="toctree-l1"><a class="reference internal" href="mergepairs.html">mergepairs</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">msa</a></li>
<li class="toctree-l1"><a class="reference internal" href="otu.html">otu</a></li>
<li class="toctree-l1"><a class="reference internal" href="root.html">root</a></li>
<li class="toctree-l1"><a class="reference internal" href="split.html">split</a></li>
<li class="toctree-l1"><a class="reference internal" href="stats.html">stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="tablebar.html">tablebar</a></li>
<li class="toctree-l1"><a class="reference internal" href="tablerare.html">tablerare</a></li>
<li class="toctree-l1"><a class="reference internal" href="tablestats.html">tablestats</a></li>
<li class="toctree-l1"><a class="reference internal" href="tabletotax.html">tabletotax</a></li>
<li class="toctree-l1"><a class="reference internal" href="tobiom.html">tobiom</a></li>
<li class="toctree-l1"><a class="reference internal" href="tree.html">tree</a></li>
<li class="toctree-l1"><a class="reference internal" href="trim.html">trim</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../index.html">micca</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../index.html">Docs</a> &raquo;</li>
      
    <li>msa</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../_sources/commands/msa.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="msa">
<h1>msa<a class="headerlink" href="#msa" title="Permalink to this headline">Â¶</a></h1>
<div class="highlight-console"><div class="highlight"><pre><span></span><span class="go">usage: micca msa [-h] -i FILE -o FILE [-m {muscle,nast}]</span>
<span class="go">                [--muscle-maxiters MUSCLE_MAXITERS] [--nast-template FILE]</span>
<span class="go">                [--nast-id NAST_ID] [--nast-threads NAST_THREADS]</span>
<span class="go">                [--nast-mincov NAST_MINCOV] [--nast-strand {both,plus}]</span>
<span class="go">                [--nast-notaligned FILE] [--nast-hits FILE] [--nast-nofilter]</span>
<span class="go">                [--nast-notrim]</span>

<span class="go">micca msa performs a multiple sequence alignment (MSA) on the input</span>
<span class="go">file in FASTA format. micca msa provides two approaches for MSA:</span>

<span class="go">* MUSCLE (doi: 10.1093/nar/gkh340). It is one of the most widely-used</span>
<span class="go">multiple sequence alignment software;</span>

<span class="go">* Nearest Alignment Space Termination (NAST) (doi:</span>
<span class="go">10.1093/nar/gkl244). MICCA provides a very fast and memory</span>
<span class="go">efficient implementation of the NAST algorithm. The algorithm is</span>
<span class="go">based on VSEARCH (https://github.com/torognes/vsearch). It requires</span>
<span class="go">a pre-aligned database of sequences (--nast-template). For 16S</span>
<span class="go">data, a good template file is the Greengenes Core Set</span>
<span class="go">(http://greengenes.lbl.gov/Download/Sequence_Data/Fasta_data_files/</span>
<span class="go">core_set_aligned.fasta.imputed).</span>

<span class="go">optional arguments:</span>
<span class="go">-h, --help            show this help message and exit</span>

<span class="go">arguments:</span>
<span class="go">-i FILE, --input FILE</span>
<span class="go">                        input FASTA file (required).</span>
<span class="go">-o FILE, --output FILE</span>
<span class="go">                        output MSA file in FASTA format (required).</span>
<span class="go">-m {muscle,nast}, --method {muscle,nast}</span>
<span class="go">                        multiple sequence alignment method (default muscle).</span>

<span class="go">MUSCLE specific options:</span>
<span class="go">--muscle-maxiters MUSCLE_MAXITERS</span>
<span class="go">                        maximum number of MUSCLE iterations. Set to 2 for a</span>
<span class="go">                        good compromise between speed and accuracy (&gt;=1</span>
<span class="go">                        default 16).</span>

<span class="go">NAST specific options:</span>
<span class="go">--nast-template FILE  multiple sequence alignment template file in FASTA</span>
<span class="go">                        format.</span>
<span class="go">--nast-id NAST_ID     sequence identity threshold to consider a sequence a</span>
<span class="go">                        match (0.0 to 1.0, default 0.75).</span>
<span class="go">--nast-threads NAST_THREADS</span>
<span class="go">                        number of threads to use (1 to 256, default 1).</span>
<span class="go">--nast-mincov NAST_MINCOV</span>
<span class="go">                        reject sequence if the fraction of alignment to the</span>
<span class="go">                        template sequence is lower than MINCOV. This parameter</span>
<span class="go">                        prevents low-coverage alignments at the end of the</span>
<span class="go">                        sequences (default 0.75).</span>
<span class="go">--nast-strand {both,plus}</span>
<span class="go">                        search both strands or the plus strand only (default</span>
<span class="go">                        both).</span>
<span class="go">--nast-notaligned FILE</span>
<span class="go">                        write not aligned sequences in FASTA format.</span>
<span class="go">--nast-hits FILE      write hits on a TAB delimited file with the query</span>
<span class="go">                        sequence id, the template sequence id and the</span>
<span class="go">                        identity.</span>
<span class="go">--nast-nofilter       do not remove positions which are gaps in every</span>
<span class="go">                        sequenceces (useful if you want to apply a Lane mask</span>
<span class="go">                        filter before the tree inference).</span>
<span class="go">--nast-notrim         force to align the entire candidate sequence (i.e. do</span>
<span class="go">                        not trim the candidate sequence to that which is bound</span>
<span class="go">                        by the beginning and end points of of the alignment</span>
<span class="go">                        span</span>

<span class="go">Examples</span>

<span class="go">De novo MSA using MUSCLE:</span>

<span class="go">    micca msa -i input.fasta -o msa.fasta</span>

<span class="go">Template-based MSA using NAST, the Greengenes alignment as</span>
<span class="go">template (clustered at 97% similarity) 4 threads and a sequence</span>
<span class="go">identity threshold of 75%:</span>

<span class="go">    micca msa -i input.fasta -o msa.fasta -m nast --nast-threads 4 \</span>
<span class="go">    --nast-template greengenes_2013_05/rep_set_aligned/97_otus.fasta</span>
</pre></div>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="otu.html" class="btn btn-neutral float-right" title="otu" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="mergepairs.html" class="btn btn-neutral" title="mergepairs" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, micca development team.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'1.6.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>