

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>otu &mdash; micca 1.6.0 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  

  

  
        <link rel="index" title="Index"
              href="../genindex.html"/>
        <link rel="search" title="Search" href="../search.html"/>
    <link rel="top" title="micca 1.6.0 documentation" href="../index.html"/>
        <link rel="next" title="root" href="root.html"/>
        <link rel="prev" title="msa" href="msa.html"/> 

  
  <script src="../_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../index.html" class="icon icon-home"> micca
          

          
          </a>

          
            
            
              <div class="version">
                1.6
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../install.html">Install</a></li>
<li class="toctree-l1"><a class="reference internal" href="../run.html">Run micca in 6 steps</a></li>
<li class="toctree-l1"><a class="reference internal" href="../databases.html">Supported databases</a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../singleend.html">Single-end sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../pairedend.html">Paired-end sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../phyloseq.html">Import and analyze the data into R and phyloseq</a></li>
<li class="toctree-l1"><a class="reference internal" href="../table.html">Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="../picrust.html">Picking OTUs for use in PICRUSt</a></li>
</ul>
<p class="caption"><span class="caption-text">In Depth</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../filtering.html">Quality filtering strategy in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otu.html">OTU picking in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="../formats.html">Supported file formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="../changes.html">Changes</a></li>
</ul>
<p class="caption"><span class="caption-text">Command reference</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="classify.html">classify</a></li>
<li class="toctree-l1"><a class="reference internal" href="convert.html">convert</a></li>
<li class="toctree-l1"><a class="reference internal" href="filter.html">filter</a></li>
<li class="toctree-l1"><a class="reference internal" href="filterstats.html">filterstats</a></li>
<li class="toctree-l1"><a class="reference internal" href="merge.html">merge</a></li>
<li class="toctree-l1"><a class="reference internal" href="mergepairs.html">mergepairs</a></li>
<li class="toctree-l1"><a class="reference internal" href="msa.html">msa</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">otu</a></li>
<li class="toctree-l1"><a class="reference internal" href="root.html">root</a></li>
<li class="toctree-l1"><a class="reference internal" href="split.html">split</a></li>
<li class="toctree-l1"><a class="reference internal" href="stats.html">stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="tablebar.html">tablebar</a></li>
<li class="toctree-l1"><a class="reference internal" href="tablerare.html">tablerare</a></li>
<li class="toctree-l1"><a class="reference internal" href="tablestats.html">tablestats</a></li>
<li class="toctree-l1"><a class="reference internal" href="tabletotax.html">tabletotax</a></li>
<li class="toctree-l1"><a class="reference internal" href="tobiom.html">tobiom</a></li>
<li class="toctree-l1"><a class="reference internal" href="tree.html">tree</a></li>
<li class="toctree-l1"><a class="reference internal" href="trim.html">trim</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../index.html">micca</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          

 



<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../index.html">Docs</a> &raquo;</li>
      
    <li>otu</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../_sources/commands/otu.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="otu">
<h1>otu<a class="headerlink" href="#otu" title="Permalink to this headline">Â¶</a></h1>
<div class="highlight-console"><div class="highlight"><pre><span></span><span class="go">usage: micca otu [-h] -i FILE [-o DIR] [-r FILE]</span>
<span class="go">                [-m {denovo_greedy,denovo_swarm,closed_ref,open_ref}] [-d ID]</span>
<span class="go">                [-n MINCOV] [-t THREADS] [-c] [-g {dgc,agc}] [-s MINSIZE]</span>
<span class="go">                [-a {both,plus}] [--swarm-differences SWARM_DIFFERENCES]</span>
<span class="go">                [--swarm-fastidious]</span>

<span class="go">micca otu assigns similar sequences (marker genes such as 16S rRNA and</span>
<span class="go">the fungal ITS region) to operational taxonomic units (OTUs).</span>

<span class="go">Trimming the sequences to a fixed position before clustering is</span>
<span class="go">*strongly recommended* when they cover partial amplicons or if quality</span>
<span class="go">deteriorates towards the end (common when you have long amplicons and</span>
<span class="go">single-end sequencing).</span>

<span class="go">Removing ambiguous nucleotides &#39;N&#39; (with the option --maxns 0 in micca</span>
<span class="go">filter) is mandatory if you use the de novo swarm clustering method.</span>

<span class="go">micca otu provides the following protocols:</span>

<span class="go">* de novo greedy clustering (denovo_greedy): sequences are clustered</span>
<span class="go">without relying on an external reference database, using an</span>
<span class="go">approach similar to the UPARSE pipeline (doi: 10.1038/nmeth.2604):</span>
<span class="go">i) predict sequence abundances of each sequence by dereplication; ii)</span>
<span class="go">greedy clustering; iii) remove chimeric sequences (de novo, optional)</span>
<span class="go">from the representatives; iv) map sequences to the representatives.</span>

<span class="go">* de novo swarm (denovo_swarm): sequences are clustered without relying</span>
<span class="go">on an external reference database, using swarm (doi:</span>
<span class="go">10.7717/peerj.593, doi: 10.7717/peerj.1420,</span>
<span class="go">https://github.com/torognes/swarm): i) predict sequence abundances of</span>
<span class="go">each sequence by dereplication; ii) swarm clustering; iii) remove</span>
<span class="go">chimeric sequences (de novo, optional) from the representatives.</span>

<span class="go">* closed-reference clustering (closed_ref): sequences are clustered</span>
<span class="go">against an external reference database and reads that could not be</span>
<span class="go">matched are discarded.</span>

<span class="go">* open-reference clustering (open_ref): sequences are clustered</span>
<span class="go">against an external reference database and reads that could not be</span>
<span class="go">matched are clustered with the &#39;de novo greedy&#39; protocol.</span>

<span class="go">Outputs:</span>

<span class="go">* otutable.txt: OTU x sample, TAB-separated OTU table file,</span>
<span class="go">containing the number of times an OTU is found in each sample.</span>

<span class="go">* otus.fasta: FASTA file containing the representative sequences (OTUs);</span>

<span class="go">* otuids.txt: OTU ids to original sequence ids (tab-delimited text file)</span>

<span class="go">* hits.txt: three-columns, TAB-separated file:</span>

<span class="go">1. matching sequence</span>
<span class="go">2. representative (seed)</span>
<span class="go">3. identity (if available, else &#39;*&#39;), defined as:</span>

<span class="go">            matching columns</span>
<span class="go">    -------------------------------- ;</span>
<span class="go">    alignment length - terminal gaps</span>

<span class="go">* otuschim.fasta (only for &#39;denovo_greedy&#39;, &#39;denovo_swarm&#39; and</span>
<span class="go">&#39;open_ref&#39; when --rmchim is specified): FASTA file containing the</span>
<span class="go">chimeric otus;</span>

<span class="go">optional arguments:</span>
<span class="go">-h, --help            show this help message and exit</span>

<span class="go">arguments:</span>
<span class="go">-i FILE, --input FILE</span>
<span class="go">                        input fasta file (required).</span>
<span class="go">-o DIR, --output DIR  output directory (default .).</span>
<span class="go">-r FILE, --ref FILE   reference sequences in fasta format, required for</span>
<span class="go">                        &#39;closed_ref&#39; and &#39;open_ref&#39; clustering methods.</span>
<span class="go">-m {denovo_greedy,denovo_swarm,closed_ref,open_ref}, --method {denovo_greedy,denovo_swarm,closed_ref,open_ref}</span>
<span class="go">                        clustering method (default denovo_greedy)</span>
<span class="go">-d ID, --id ID        sequence identity threshold (for &#39;denovo_greedy&#39;,</span>
<span class="go">                        &#39;closed_ref&#39; and &#39;open_ref&#39;, 0.0 to 1.0, default</span>
<span class="go">                        0.97).</span>
<span class="go">-n MINCOV, --mincov MINCOV</span>
<span class="go">                        reject sequence if the fraction of alignment to the</span>
<span class="go">                        reference sequence is lower than MINCOV. This</span>
<span class="go">                        parameter prevents low-coverage alignments at the end</span>
<span class="go">                        of the sequences (for &#39;closed_ref&#39; and &#39;open_ref&#39;</span>
<span class="go">                        clustering methods, default 0.75).</span>
<span class="go">-t THREADS, --threads THREADS</span>
<span class="go">                        number of threads to use (1 to 256, default 1).</span>
<span class="go">-c, --rmchim          remove chimeric sequences (for &#39;denovo_greedy&#39;,</span>
<span class="go">                        &#39;denovo_swarm&#39; and &#39;open_ref&#39; OTU picking methods).</span>
<span class="go">-g {dgc,agc}, --greedy {dgc,agc}</span>
<span class="go">                        greedy clustering strategy, distance (DGC) or</span>
<span class="go">                        abundance-based (AGC) (for &#39;denovo_greedy&#39; and</span>
<span class="go">                        &#39;open_ref&#39; clustering methods) (default dgc).</span>
<span class="go">-s MINSIZE, --minsize MINSIZE</span>
<span class="go">                        discard sequences with an abundance value smaller than</span>
<span class="go">                        MINSIZE after dereplication (&gt;=1, default 2).</span>
<span class="go">                        Recommended value is 2 (i.e. discard singletons) for</span>
<span class="go">                        &#39;denovo_greedy&#39; and &#39;open_ref&#39;, 1 (do not discard</span>
<span class="go">                        anything) for &#39;denovo_swarm&#39;.</span>
<span class="go">-a {both,plus}, --strand {both,plus}</span>
<span class="go">                        search both strands or the plus strand only (for</span>
<span class="go">                        &#39;closed_ref&#39; and &#39;open_ref&#39; clustering methods,</span>
<span class="go">                        default both).</span>

<span class="go">Swarm specific options:</span>
<span class="go">--swarm-differences SWARM_DIFFERENCES</span>
<span class="go">                        maximum number of differences allowed between two</span>
<span class="go">                        amplicons. Commonly used d values are 1 (linear</span>
<span class="go">                        complexity algorithm), 2 or 3, rarely higher. (&gt;=0,</span>
<span class="go">                        default 1).</span>
<span class="go">--swarm-fastidious    when working with SWARM_DIFFERENCES=1, perform a</span>
<span class="go">                        second clustering pass to reduce the number of small</span>
<span class="go">                        OTUs (recommended option).</span>

<span class="go">Examples</span>

<span class="go">De novo clustering with a 97% similarity threshold and remove</span>
<span class="go">chimeric OTUs:</span>

<span class="go">    micca otu -i input.fasta --method denovo_greedy --id 0.97 -c</span>

<span class="go">Open-reference OTU picking protocol with a 97% similarity</span>
<span class="go">threshold, without removing chimeras in the de novo protocol step</span>
<span class="go">and using 8 threads:</span>

<span class="go">    micca otu -i input.fasta --method open_ref --threads 8 --id 0.97 \</span>
<span class="go">    --ref greengenes_2013_05/rep_set/97_otus.fasta</span>

<span class="go">De novo swarm clustering with the protocol suggested by the authors</span>
<span class="go">using 4 threads (see https://github.com/torognes/swarm and</span>
<span class="go">https://github.com/torognes/swarm/wiki):</span>

<span class="go">    micca otu -i input.fasta --method denovo_swarm --threads 4 \</span>
<span class="go">    --swarm-fastidious --rmchim --minsize 1</span>
</pre></div>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="root.html" class="btn btn-neutral float-right" title="root" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="msa.html" class="btn btn-neutral" title="msa" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2017, micca development team.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'1.6.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>