

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Picking OTUs for use in PICRUSt &mdash; micca 1.1.0 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
    <link rel="top" title="micca 1.1.0 documentation" href="index.html"/>
        <link rel="next" title="Quality filtering strategy in micca" href="filtering.html"/>
        <link rel="prev" title="Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca" href="table.html"/> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> micca
          

          
          </a>

          
            
            
              <div class="version">
                1.1.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="install.html">Download and install</a></li>
<li class="toctree-l1"><a class="reference internal" href="run.html">Run micca in 6 steps</a></li>
<li class="toctree-l1"><a class="reference internal" href="databases.html">Supported databases</a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="singleend.html">Single-end sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="pairedend.html">Paired-end sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="phyloseq.html">Import and analyze the data into R and phyloseq</a></li>
<li class="toctree-l1"><a class="reference internal" href="table.html">Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="">Picking OTUs for use in PICRUSt</a></li>
</ul>
<p class="caption"><span class="caption-text">In Depth</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="filtering.html">Quality filtering strategy in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="otu.html">OTU picking in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="formats.html">Supported file formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="changes.html">CHANGES</a></li>
</ul>
<p class="caption"><span class="caption-text">Command reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="commands/classify.html">classify</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/convert.html">convert</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/filter.html">filter</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/filterstats.html">filterstats</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/merge.html">merge</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/mergepairs.html">mergepairs</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/msa.html">msa</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/otu.html">otu</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/root.html">root</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/split.html">split</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/stats.html">stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tablebar.html">tablebar</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tablerare.html">tablerare</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tablestats.html">tablestats</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tabletotax.html">tabletotax</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tobiom.html">tobiom</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/tree.html">tree</a></li>
<li class="toctree-l1"><a class="reference internal" href="commands/trim.html">trim</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="index.html">micca</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          





<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="index.html">Docs</a> &raquo;</li>
      
    <li>Picking OTUs for use in PICRUSt</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/picrust.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="picking-otus-for-use-in-picrust">
<h1>Picking OTUs for use in PICRUSt<a class="headerlink" href="#picking-otus-for-use-in-picrust" title="Permalink to this headline">Â¶</a></h1>
<p><a class="reference external" href="http://picrust.github.io/picrust/">PICRUSt</a> (doi: 10.1038/nbt.2676)
is a software designed to predict metagenome functional content from
marker gene (e.g., 16S rRNA) surveys and full genomes. This tutorial
covers how to pick OTUs from 16S rRNA sequences data to use with
PICRUSt.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Requires <a class="reference internal" href="singleend.html#singleend-quality-filtering"><span>Quality filtering</span></a> in <a class="reference internal" href="singleend.html"><em>Single-end sequencing</em></a> to
be done and the PICRUSt software to be installed in your system.
Warning: PICRUSt 1.0.0 requires the biom-format package v1.3.1 to
be installed in your system (from the command line run: <code class="docutils literal"><span class="pre">pip</span>
<span class="pre">install</span> <span class="pre">biom-format==1.3.1</span></code>, for more information see
<a class="reference external" href="http://biom-format.org/">http://biom-format.org/</a>).</p>
</div>
<p>PICRUSt requires an <a class="reference internal" href="otu.html#otu-closed-reference"><span>Closed-reference</span></a> OTU table computed
against the Greengenes reference (clustered at 97% identity).
Download the reference database (Greengenes, version 2013/05),
clustered at 97% identity:</p>
<div class="highlight-sh"><div class="highlight"><pre>wget ftp://ftp.fmach.it/metagenomics/micca/dbs/gg_2013_05.tar.gz
tar -zxvf gg_2013_05.tar.gz
</pre></div>
</div>
<p>Run the micca closed-reference protocol:</p>
<div class="highlight-sh"><div class="highlight"><pre>micca otu -m closed_ref -i filtered.fasta -o closed_ref_otus -r 97_otus.fasta -d 0.97 -t 4
<span class="nb">cd</span> closed_ref_otus
</pre></div>
</div>
<p>Report the sample summary:</p>
<div class="highlight-sh"><div class="highlight"><pre>micca tablestats -i otutable.txt -o tablestats
head tablestats/tablestats_samplesumm.txt

Sample       Depth   NOTU    NSingle
Mw_03        <span class="m">1084</span>    <span class="m">132</span>     39
Mw_06        <span class="m">1387</span>    <span class="m">122</span>     27
Mw_11        <span class="m">1485</span>    <span class="m">155</span>     44
Mw_07        <span class="m">1528</span>    <span class="m">150</span>     36
Mw_01        <span class="m">1537</span>    <span class="m">143</span>     35
Mw_15        <span class="m">1565</span>    <span class="m">144</span>     35
Mw_14        <span class="m">1610</span>    <span class="m">149</span>     42
Mw_02        <span class="m">1670</span>    <span class="m">143</span>     43
Mw_12        <span class="m">1710</span>    <span class="m">153</span>     54
</pre></div>
</div>
<p>Rarefy the OTU table for the PICRUSt analysis is always a good idea
(see
<a class="reference external" href="https://groups.google.com/forum/#!topic/picrust-users/ev5uZGUIPrQ">https://groups.google.com/forum/#!topic/picrust-users/ev5uZGUIPrQ</a>), so
we will rarefy the table at 1084 sequences per sample using
<a class="reference internal" href="commands/tablerare.html"><em>tablerare</em></a>:</p>
<div class="highlight-sh"><div class="highlight"><pre>micca tablerare -i otutable.txt -o otutable_rare.txt -d 1084
</pre></div>
</div>
<p>Convert the rarefied OTU table into the BIOM format replacing the OTU
IDs with the original sequence IDs using the
<a class="reference internal" href="commands/tobiom.html"><em>tobiom</em></a> command:</p>
<div class="highlight-sh"><div class="highlight"><pre>micca tobiom -i otutable_rare.txt -o tables.biom -u otuids.txt
</pre></div>
</div>
<p>Normalize the OTU table by dividing each OTU by the known/predicted
16S copy number abundance:</p>
<div class="highlight-sh"><div class="highlight"><pre>normalize_by_copy_number.py -i tables.biom -o normalized_otus.biom
</pre></div>
</div>
<p>Create the final metagenome functional predictions:</p>
<div class="highlight-sh"><div class="highlight"><pre>predict_metagenomes.py -i normalized_otus.biom -o metagenome_predictions.biom
</pre></div>
</div>
<p>Now you can analyze the PICRUSt predicted metagenome:
<a class="reference external" href="http://picrust.github.io/picrust/tutorials/downstream_analysis.html#downstream-analysis-guide">http://picrust.github.io/picrust/tutorials/downstream_analysis.html#downstream-analysis-guide</a>.</p>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="filtering.html" class="btn btn-neutral float-right" title="Quality filtering strategy in micca" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="table.html" class="btn btn-neutral" title="Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2016, micca development team.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'1.1.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>