

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>otu &mdash; micca 1.4.0 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  

  

  
    <link rel="top" title="micca 1.4.0 documentation" href="../index.html"/>
        <link rel="next" title="root" href="root.html"/>
        <link rel="prev" title="msa" href="msa.html"/> 

  
  <script src="../_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../index.html" class="icon icon-home"> micca
          

          
          </a>

          
            
            
              <div class="version">
                1.4.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../install.html">Install</a></li>
<li class="toctree-l1"><a class="reference internal" href="../run.html">Run micca in 6 steps</a></li>
<li class="toctree-l1"><a class="reference internal" href="../databases.html">Supported databases</a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../singleend.html">Single-end sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../pairedend.html">Paired-end sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../phyloseq.html">Import and analyze the data into R and phyloseq</a></li>
<li class="toctree-l1"><a class="reference internal" href="../table.html">Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="../picrust.html">Picking OTUs for use in PICRUSt</a></li>
</ul>
<p class="caption"><span class="caption-text">In Depth</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../filtering.html">Quality filtering strategy in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otu.html">OTU picking in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="../formats.html">Supported file formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="../changes.html">CHANGES</a></li>
</ul>
<p class="caption"><span class="caption-text">Command reference</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="classify.html">classify</a></li>
<li class="toctree-l1"><a class="reference internal" href="convert.html">convert</a></li>
<li class="toctree-l1"><a class="reference internal" href="filter.html">filter</a></li>
<li class="toctree-l1"><a class="reference internal" href="filterstats.html">filterstats</a></li>
<li class="toctree-l1"><a class="reference internal" href="merge.html">merge</a></li>
<li class="toctree-l1"><a class="reference internal" href="mergepairs.html">mergepairs</a></li>
<li class="toctree-l1"><a class="reference internal" href="msa.html">msa</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="">otu</a></li>
<li class="toctree-l1"><a class="reference internal" href="root.html">root</a></li>
<li class="toctree-l1"><a class="reference internal" href="split.html">split</a></li>
<li class="toctree-l1"><a class="reference internal" href="stats.html">stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="tablebar.html">tablebar</a></li>
<li class="toctree-l1"><a class="reference internal" href="tablerare.html">tablerare</a></li>
<li class="toctree-l1"><a class="reference internal" href="tablestats.html">tablestats</a></li>
<li class="toctree-l1"><a class="reference internal" href="tabletotax.html">tabletotax</a></li>
<li class="toctree-l1"><a class="reference internal" href="tobiom.html">tobiom</a></li>
<li class="toctree-l1"><a class="reference internal" href="tree.html">tree</a></li>
<li class="toctree-l1"><a class="reference internal" href="trim.html">trim</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../index.html">micca</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          





<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../index.html">Docs</a> &raquo;</li>
      
    <li>otu</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../_sources/commands/otu.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="otu">
<h1>otu<a class="headerlink" href="#otu" title="Permalink to this headline">Â¶</a></h1>
<div class="highlight-text"><div class="highlight"><pre>$ micca otu --help
usage: micca otu [-h] -i FILE [-o DIR] [-r FILE]
                 [-m {denovo_greedy,denovo_swarm,closed_ref,open_ref}] [-d ID]
                 [-n MINCOV] [-t THREADS] [-c] [-g {dgc,agc}] [-s MINSIZE]
                 [-a {both,plus}] [--swarm-differences SWARM_DIFFERENCES]
                 [--swarm-fastidious]

micca otu assigns similar sequences (marker genes such as 16S rRNA and
the fungal ITS region) to operational taxonomic units (OTUs).

Trimming the sequences to a fixed position before clustering is
*strongly recommended* when they cover partial amplicons or if quality
deteriorates towards the end (common when you have long amplicons and
single-end sequencing).

Removing ambiguous nucleotides &#39;N&#39; (with the option --maxns 0 in micca
filter) is mandatory if you use the de novo swarm clustering method.

micca otu provides the following protocols:

* de novo greedy clustering (denovo_greedy): sequences are clustered
  without relying on an external reference database, using an
  approach similar to the UPARSE pipeline (doi: 10.1038/nmeth.2604):
  i) predict sequence abundances of each sequence by dereplication; ii)
  greedy clustering; iii) remove chimeric sequences (de novo, optional)
  from the representatives; iv) map sequences to the representatives.

* de novo swarm (denovo_swarm): sequences are clustered without relying
  on an external reference database, using swarm (doi:
  10.7717/peerj.593, doi: 10.7717/peerj.1420,
  https://github.com/torognes/swarm): i) predict sequence abundances of
  each sequence by dereplication; ii) swarm clustering; iii) remove
  chimeric sequences (de novo, optional) from the representatives.

* closed-reference clustering (closed_ref): sequences are clustered
  against an external reference database and reads that could not be
  matched are discarded.

* open-reference clustering (open_ref): sequences are clustered
  against an external reference database and reads that could not be
  matched are clustered with the &#39;de novo greedy&#39; protocol.

Outputs:

* otutable.txt: OTU x sample, TAB-separated OTU table file,
  containing the number of times an OTU is found in each sample.

* otus.fasta: FASTA file containing the representative sequences (OTUs);

* otuids.txt: OTU ids to original sequence ids (tab-delimited text file)

* hits.txt: three-columns, TAB-separated file:

  1. matching sequence
  2. representative (seed)
  3. identity (if available, else &#39;*&#39;), defined as:

              matching columns
     -------------------------------- ;
     alignment length - terminal gaps

* otuschim.fasta (only for &#39;denovo_greedy&#39;, &#39;denovo_swarm&#39; and
  &#39;open_ref&#39; when --rmchim is specified): FASTA file containing the
  chimeric otus;

optional arguments:
  -h, --help            show this help message and exit

arguments:
  -i FILE, --input FILE
                        input fasta file (required).
  -o DIR, --output DIR  output directory (default .).
  -r FILE, --ref FILE   reference sequences in fasta format, required for
                        &#39;closed_ref&#39; and &#39;open_ref&#39; clustering methods.
  -m {denovo_greedy,denovo_swarm,closed_ref,open_ref}, --method {denovo_greedy,denovo_swarm,closed_ref,open_ref}
                        clustering method (default denovo_greedy)
  -d ID, --id ID        sequence identity threshold (for &#39;denovo_greedy&#39;,
                        &#39;closed_ref&#39; and &#39;open_ref&#39;, 0.0 to 1.0, default
                        0.97).
  -n MINCOV, --mincov MINCOV
                        reject sequence if the fraction of alignment to the
                        reference sequence is lower than MINCOV. This
                        parameter prevents low-coverage alignments at the end
                        of the sequences (for &#39;closed_ref&#39; and &#39;open_ref&#39;
                        clustering methods, default 0.75).
  -t THREADS, --threads THREADS
                        number of threads to use (1 to 256, default 1).
  -c, --rmchim          remove chimeric sequences (for &#39;denovo_greedy&#39;,
                        &#39;denovo_swarm&#39; and &#39;open_ref&#39; OTU picking methods).
  -g {dgc,agc}, --greedy {dgc,agc}
                        greedy clustering strategy, distance (DGC) or
                        abundance-based (AGC) (for &#39;denovo_greedy&#39; and
                        &#39;open_ref&#39; clustering methods) (default dgc).
  -s MINSIZE, --minsize MINSIZE
                        discard sequences with an abundance value smaller than
                        MINSIZE after dereplication (&gt;=1, default 2).
                        Recommended value is 2 (i.e. discard singletons) for
                        &#39;denovo_greedy&#39; and &#39;open_ref&#39;, 1 (do not discard
                        anything) for &#39;denovo_swarm&#39;.
  -a {both,plus}, --strand {both,plus}
                        search both strands or the plus strand only (for
                        &#39;closed_ref&#39; and &#39;open_ref&#39; clustering methods,
                        default both).

Swarm specific options:
  --swarm-differences SWARM_DIFFERENCES
                        maximum number of differences allowed between two
                        amplicons. Commonly used d values are 1 (linear
                        complexity algorithm), 2 or 3, rarely higher. (&gt;=0,
                        default 1).
  --swarm-fastidious    when working with SWARM_DIFFERENCES=1, perform a
                        second clustering pass to reduce the number of small
                        OTUs (recommended option).

Examples

De novo clustering with a 97% similarity threshold and remove
chimeric OTUs:

    micca otu -i input.fasta --method denovo_greedy --id 0.97 -c

Open-reference OTU picking protocol with a 97% similarity
threshold, without removing chimeras in the de novo protocol step
and using 8 threads:

    micca otu -i input.fasta --method open_ref --threads 8 --id 0.97 \
    --ref greengenes_2013_05/rep_set/97_otus.fasta

De novo swarm clustering with the protocol suggested by the authors
using 4 threads (see https://github.com/torognes/swarm and
https://github.com/torognes/swarm/wiki):

    micca otu -i input.fasta --method denovo_swarm --threads 4 \
    --swarm-fastidious --rmchim --minsize 1
</pre></div>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="root.html" class="btn btn-neutral float-right" title="root" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="msa.html" class="btn btn-neutral" title="msa" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2016, micca development team.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'1.4.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>