

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>classify &mdash; micca 1.3.0 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  

  

  
    <link rel="top" title="micca 1.3.0 documentation" href="../index.html"/>
        <link rel="next" title="convert" href="convert.html"/>
        <link rel="prev" title="CHANGES" href="../changes.html"/> 

  
  <script src="../_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../index.html" class="icon icon-home"> micca
          

          
          </a>

          
            
            
              <div class="version">
                1.3.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <p class="caption"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../install.html">Download and install</a></li>
<li class="toctree-l1"><a class="reference internal" href="../run.html">Run micca in 6 steps</a></li>
<li class="toctree-l1"><a class="reference internal" href="../databases.html">Supported databases</a></li>
</ul>
<p class="caption"><span class="caption-text">Tutorials</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../singleend.html">Single-end sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../pairedend.html">Paired-end sequencing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../phyloseq.html">Import and analyze the data into R and phyloseq</a></li>
<li class="toctree-l1"><a class="reference internal" href="../table.html">Compute basic OTU table statistics, rarefy and summarize OTU tables by taxa using micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="../picrust.html">Picking OTUs for use in PICRUSt</a></li>
</ul>
<p class="caption"><span class="caption-text">In Depth</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../filtering.html">Quality filtering strategy in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otu.html">OTU picking in micca</a></li>
<li class="toctree-l1"><a class="reference internal" href="../formats.html">Supported file formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="../changes.html">CHANGES</a></li>
</ul>
<p class="caption"><span class="caption-text">Command reference</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="">classify</a></li>
<li class="toctree-l1"><a class="reference internal" href="convert.html">convert</a></li>
<li class="toctree-l1"><a class="reference internal" href="filter.html">filter</a></li>
<li class="toctree-l1"><a class="reference internal" href="filterstats.html">filterstats</a></li>
<li class="toctree-l1"><a class="reference internal" href="merge.html">merge</a></li>
<li class="toctree-l1"><a class="reference internal" href="mergepairs.html">mergepairs</a></li>
<li class="toctree-l1"><a class="reference internal" href="msa.html">msa</a></li>
<li class="toctree-l1"><a class="reference internal" href="otu.html">otu</a></li>
<li class="toctree-l1"><a class="reference internal" href="root.html">root</a></li>
<li class="toctree-l1"><a class="reference internal" href="split.html">split</a></li>
<li class="toctree-l1"><a class="reference internal" href="stats.html">stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="tablebar.html">tablebar</a></li>
<li class="toctree-l1"><a class="reference internal" href="tablerare.html">tablerare</a></li>
<li class="toctree-l1"><a class="reference internal" href="tablestats.html">tablestats</a></li>
<li class="toctree-l1"><a class="reference internal" href="tabletotax.html">tabletotax</a></li>
<li class="toctree-l1"><a class="reference internal" href="tobiom.html">tobiom</a></li>
<li class="toctree-l1"><a class="reference internal" href="tree.html">tree</a></li>
<li class="toctree-l1"><a class="reference internal" href="trim.html">trim</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../index.html">micca</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          





<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../index.html">Docs</a> &raquo;</li>
      
    <li>classify</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../_sources/commands/classify.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="classify">
<h1>classify<a class="headerlink" href="#classify" title="Permalink to this headline">Â¶</a></h1>
<div class="highlight-text"><div class="highlight"><pre>$ micca classify --help
usage: micca classify [-h] -i FILE -o FILE [-m {cons,rdp,otuid}] [-r FILE]
                      [-x FILE] [--cons-id CONS_ID]
                      [--cons-maxhits CONS_MAXHITS]
                      [--cons-minfrac CONS_MINFRAC]
                      [--cons-mincov CONS_MINCOV] [--cons-strand {both,plus}]
                      [--cons-threads THREADS]
                      [--rdp-gene {16srrna,fungallsu,fungalits_warcup,fungalits_unite}]
                      [--rdp-maxmem GB] [--rdp-minconf RDP_MINCONF]

micca classify assigns taxonomy for each sequence in the input file
and provides three methods for classification:

* VSEARCH-based consensus classifier (cons): input sequences are
  searched in the reference database with VSEARCH
  (https://github.com/torognes/vsearch). For each query sequence the
  method retrives up to &#39;cons-maxhits&#39; hits (i.e. identity &gt;=
  &#39;cons-id&#39;). Then, the most specific taxonomic label that is
  associated with at least &#39;cons-minfrac&#39; of the hits is
  assigned. The method is similar to the UCLUST-based consensus
  taxonomy assigner presented in doi: 10.7287/peerj.preprints.934v2
  and available in QIIME.

* RDP classifier (rdp): only RDP classifier version &gt;= 2.8 is
  supported (doi:10.1128/AEM.00062-07). In order to use this
  classifier RDP must be installed (download at
  http://sourceforge.net/projects/rdp-classifier/files/rdp-classifier/)
  and the RDPPATH environment variable setted. The available
  databases (--rdp-gene) are:

  - 16S (16srrna)
  - Fungal LSU (28S) (fungallsu)
  - Warcup ITS (fungalits_warcup, doi: 10.3852/14-293)
  - UNITE ITS (fungalits_unite)

  For more information about the RDP classifier go to
  http://rdp.cme.msu.edu/classifier/classifier.jsp

* OTU ID classifier (otuid): simply perform a sequence ID matching
  with the reference taxonomy file. Recommended strategy when the
  closed reference clustering (--method closedref in micca-otu) was
  performed. OTU ID classifier requires a tab-delimited file where
  the first column contains the current OTU ids and the second column
  the reference taxonomy ids (see otuids.txt in micca-otu), e.g.:

  REF1[TAB]1110191
  REF2[TAB]1104777
  REF3[TAB]1078527
  ...

The input reference taxonomy file (--ref-tax) should be a
tab-delimited file where rows are either in the form:

1. SEQID[TAB]k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__;g__;
2. SEQID[TAB]Bacteria;Firmicutes;Clostridia;Clostridiales;;;
3. SEQID[TAB]Bacteria;Firmicutes;Clostridia;Clostridiales

Compatible reference database are Greengenes
(http://greengenes.secondgenome.com/downloads), QIIME-formatted SILVA
(https://www.arb-silva.de/download/archive/qiime/) and UNITE
(https://unite.ut.ee/repository.php).

The output file is a tab-delimited file where each row is in the
format:

SEQID[TAB]Bacteria;Firmicutes;Clostridia;Clostridiales

optional arguments:
  -h, --help            show this help message and exit

arguments:
  -i FILE, --input FILE
                        input FASTA file (for &#39;cons&#39; and &#39;rdp&#39;) or a tab-
                        delimited OTU ids file (for &#39;otuid&#39;) (required).
  -o FILE, --output FILE
                        output taxonomy file (required).
  -m {cons,rdp,otuid}, --method {cons,rdp,otuid}
                        classification method (default cons)
  -r FILE, --ref FILE   reference sequences in FASTA format, required for
                        &#39;cons&#39; classifier.
  -x FILE, --ref-tax FILE
                        tab-separated reference taxonomy file, required for
                        &#39;cons&#39; and &#39;otuid&#39; classifiers.

VSEARCH-based consensus classifierspecific options:
  --cons-id CONS_ID     sequence identity threshold (0.0 to 1.0, default 0.9).
  --cons-maxhits CONS_MAXHITS
                        number of hits to consider (&gt;=1, default 3).
  --cons-minfrac CONS_MINFRAC
                        for each taxonomic rank, a specific taxa will be
                        assigned if it is present in at least MINFRAC of the
                        hits (0.0 to 1.0, default 0.5).
  --cons-mincov CONS_MINCOV
                        reject sequence if the fraction of alignment to the
                        reference sequence is lower than MINCOV. This
                        parameter prevents low-coverage alignments at the end
                        of the sequences (default 0.75).
  --cons-strand {both,plus}
                        search both strands or the plus strand only (default
                        both).
  --cons-threads THREADS
                        number of threads to use (1 to 256, default 1).

RDP Classifier/Database specific options:
  --rdp-gene {16srrna,fungallsu,fungalits_warcup,fungalits_unite}
                        marker gene/region
  --rdp-maxmem GB       maximum memory size for the java virtual machine in GB
                        (default 2)
  --rdp-minconf RDP_MINCONF
                        minimum confidence value to assign taxonomy to a
                        sequence (default 0.8)

Examples

Classification of 16S sequences using the consensus classifier and
Greengenes:

    micca classify -m cons -i input.fasta -o tax.txt \
    --ref greengenes_2013_05/rep_set/97_otus.fasta \
    --ref-tax greengenes_2013_05/taxonomy/97_otu_taxonomy.txt

Classification of ITS sequences using the RDP classifier and the
UNITE database:

    micca classify -m rdp --rdp-gene fungalits_unite -i input.fasta \
    -o tax.txt

OTU ID matching after the closed reference OTU picking protocol:

    micca classify -m otuid -i otuids.txt -o tax.txt \
    --ref-tax greengenes_2013_05/taxonomy/97_otu_taxonomy.txt
</pre></div>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="convert.html" class="btn btn-neutral float-right" title="convert" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../changes.html" class="btn btn-neutral" title="CHANGES" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2016, micca development team.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'1.3.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>