

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Importing micca outputs in R and phyloseq &mdash; micca 0.1 documentation</title>
    
    <link rel="stylesheet" href="../_static/default.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="top" title="micca 0.1 documentation" href="../index.html" />
    <link rel="up" title="Tutorials" href="../tutorials.html" />
    <link rel="next" title="Command Reference" href="../command_ref.html" />
    <link rel="prev" title="16S Amplicon Reads Tutorial" href="16S.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="../command_ref.html" title="Command Reference"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="16S.html" title="16S Amplicon Reads Tutorial"
             accesskey="P">previous</a> |</li>
        <li><a href="../index.html">micca 0.1 documentation</a> &raquo;</li>
          <li><a href="../tutorials.html" accesskey="U">Tutorials</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="importing-micca-outputs-in-r-and-phyloseq">
<h1>Importing micca outputs in R and <tt class="docutils literal"><span class="pre">phyloseq</span></tt><a class="headerlink" href="#importing-micca-outputs-in-r-and-phyloseq" title="Permalink to this headline">Â¶</a></h1>
<p><tt class="docutils literal"><span class="pre">phyloseq</span></tt> is a popular R package for the analysis and visualization
of microbial data, but it does not have an official support for
<tt class="docutils literal"><span class="pre">micca</span></tt> outputs.</p>
<p>From the <a class="reference external" href="http://joey711.github.io/phyloseq/">phyloseq</a> homepage:
| The phyloseq package is a tool to import, store, analyze, and
| graphically display complex phylogenetic sequencing data that has
| already been clustered into Operational Taxonomic Units (OTUs),
| especially when there is associated sample data, phylogenetic tree,
| and/or taxonomic assignment of the OTUs.&#8221;</p>
<p>In order to import <tt class="docutils literal"><span class="pre">micca</span></tt> outputs in <tt class="docutils literal"><span class="pre">phyloseq</span></tt>, you have to
define the following function or import the file <tt class="docutils literal"><span class="pre">R/micca2ps.R</span></tt>.</p>
<div class="highlight-r"><div class="highlight"><pre>library<span class="p">(</span>phyloseq<span class="p">)</span>

import_micca <span class="o">=</span> <span class="kr">function</span><span class="p">(</span>otufilename<span class="o">=</span><span class="kc">NULL</span><span class="p">,</span> taxonomyfilename<span class="o">=</span><span class="kc">NULL</span><span class="p">,</span>
                        samplefilename<span class="o">=</span><span class="kc">NULL</span><span class="p">,</span> treefilename<span class="o">=</span><span class="kc">NULL</span><span class="p">)</span> <span class="p">{</span>
    rank <span class="o">=</span> c<span class="p">(</span><span class="s">&quot;Kingdom&quot;</span><span class="p">,</span> <span class="s">&quot;Phylum&quot;</span><span class="p">,</span> <span class="s">&quot;Class&quot;</span><span class="p">,</span> <span class="s">&quot;Order&quot;</span><span class="p">,</span> <span class="s">&quot;Family&quot;</span><span class="p">,</span> <span class="s">&quot;Genus&quot;</span><span class="p">,</span> <span class="s">&quot;Species&quot;</span><span class="p">)</span>
    argumentlist <span class="o">=</span> list<span class="p">()</span>
    <span class="kr">if</span> <span class="p">(</span><span class="o">!</span>is.null<span class="p">(</span>otufilename<span class="p">))</span> <span class="p">{</span>
        otu_frame <span class="o">=</span> read.csv<span class="p">(</span>otufilename<span class="p">,</span> sep<span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> header<span class="o">=</span><span class="k-Variable">T</span><span class="p">,</span> row.names<span class="o">=</span><span class="m">1</span><span class="p">,</span>
            check.names<span class="o">=</span><span class="k-Variable">F</span><span class="p">,</span> stringsAsFactors<span class="o">=</span><span class="k-Variable">F</span><span class="p">)</span>
        otu <span class="o">=</span> otu_table<span class="p">(</span>otu_frame<span class="p">,</span> taxa_are_rows<span class="o">=</span><span class="k-Variable">T</span><span class="p">)</span>
        argumentlist <span class="o">=</span> c<span class="p">(</span>argumentlist<span class="p">,</span> list<span class="p">(</span>otu<span class="p">))</span>
    <span class="p">}</span>
    <span class="kr">if</span> <span class="p">(</span><span class="o">!</span>is.null<span class="p">(</span>taxonomyfilename<span class="p">))</span> <span class="p">{</span>
        taxa_frame <span class="o">=</span> read.table<span class="p">(</span>taxonomyfilename<span class="p">,</span> sep<span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> header<span class="o">=</span><span class="k-Variable">F</span><span class="p">,</span>
            check.names<span class="o">=</span><span class="k-Variable">F</span><span class="p">,</span> stringsAsFactors<span class="o">=</span><span class="k-Variable">F</span><span class="p">)</span>
        taxa_split <span class="o">=</span> strsplit<span class="p">(</span>taxa_frame<span class="p">[,</span><span class="m">2</span><span class="p">],</span> <span class="s">&quot;;&quot;</span><span class="p">)</span>
        names<span class="p">(</span>taxa_split<span class="p">)</span> <span class="o">=</span> taxa_frame<span class="p">[,</span><span class="m">1</span><span class="p">]</span>
        taxa <span class="o">=</span> build_tax_table<span class="p">(</span>sapply<span class="p">(</span>taxa_split<span class="p">,</span> parse_taxonomy_default<span class="p">))</span>
        colnames<span class="p">(</span>taxa<span class="p">)</span> <span class="o">=</span> rank<span class="p">[</span><span class="m">1</span><span class="o">:</span>dim<span class="p">(</span>taxa<span class="p">)[[</span><span class="m">2</span><span class="p">]]]</span>
        argumentlist <span class="o">=</span> c<span class="p">(</span>argumentlist<span class="p">,</span> list<span class="p">(</span>taxa<span class="p">))</span>
    <span class="p">}</span>
    <span class="kr">if</span> <span class="p">(</span><span class="o">!</span>is.null<span class="p">(</span>samplefilename<span class="p">))</span> <span class="p">{</span>
        sample_frame <span class="o">=</span> read.csv<span class="p">(</span>samplefilename<span class="p">,</span> sep<span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">,</span> header<span class="o">=</span><span class="k-Variable">T</span><span class="p">,</span> row.names<span class="o">=</span><span class="m">1</span><span class="p">,</span>
            check.names<span class="o">=</span><span class="k-Variable">F</span><span class="p">,</span> stringsAsFactors<span class="o">=</span><span class="k-Variable">F</span><span class="p">)</span>
        sample <span class="o">=</span> sample_data<span class="p">(</span>sample_frame<span class="p">)</span>
        argumentlist <span class="o">=</span> c<span class="p">(</span>argumentlist<span class="p">,</span> list<span class="p">(</span>sample<span class="p">))</span>
    <span class="p">}</span>
    <span class="kr">if</span> <span class="p">(</span><span class="o">!</span>is.null<span class="p">(</span>treefilename<span class="p">))</span> <span class="p">{</span>
        tree <span class="o">=</span> read_tree<span class="p">(</span>treefilename<span class="p">)</span>
        <span class="kr">if</span> <span class="p">(</span>is.null<span class="p">(</span>tree<span class="p">))</span> <span class="p">{</span>
            warning<span class="p">(</span><span class="s">&quot;treefilename failed import. It will not be included.&quot;</span><span class="p">)</span>
        <span class="p">}</span>
        <span class="kr">else</span> <span class="p">{</span>
            argumentlist <span class="o">=</span> c<span class="p">(</span>argumentlist<span class="p">,</span> list<span class="p">(</span>tree<span class="p">))</span>
        <span class="p">}</span>
    <span class="p">}</span>
    
    do.call<span class="p">(</span><span class="s">&quot;phyloseq&quot;</span><span class="p">,</span> argumentlist<span class="p">)</span>
<span class="p">}</span>
</pre></div>
</div>
<p>Now, you can import <tt class="docutils literal"><span class="pre">micca</span></tt> outputs defined in <a class="reference internal" href="16S.html"><em>16S Amplicon Reads Tutorial</em></a>:</p>
<div class="highlight-r"><div class="highlight"><pre>library<span class="p">(</span>phyloseq<span class="p">)</span>
source<span class="p">(</span><span class="s">&quot;micca2ps.R&quot;</span><span class="p">)</span>
ps <span class="o">=</span> import_micca<span class="p">(</span>otufilename<span class="o">=</span><span class="s">&quot;otu_rdp/otu_table.txt&quot;</span><span class="p">,</span>
                  taxonomyfilename<span class="o">=</span><span class="s">&quot;otu_rdp/taxonomy.txt&quot;</span><span class="p">,</span>
                      treefilename<span class="o">=</span><span class="s">&quot;phylo_pynast/tree_rooted.tre&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="../index.html">
              <img class="logo" src="../_static/micca_logo.png" alt="Logo"/>
            </a></p>
  <h4>Previous topic</h4>
  <p class="topless"><a href="16S.html"
                        title="previous chapter">16S Amplicon Reads Tutorial</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="../command_ref.html"
                        title="next chapter">Command Reference</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="../_sources/tutorials/phyloseq.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="../command_ref.html" title="Command Reference"
             >next</a> |</li>
        <li class="right" >
          <a href="16S.html" title="16S Amplicon Reads Tutorial"
             >previous</a> |</li>
        <li><a href="../index.html">micca 0.1 documentation</a> &raquo;</li>
          <li><a href="../tutorials.html" >Tutorials</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2014, Davide Albanese, Paolo Fontana, Claudio Donati.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>